---
title: __Succession of ambrosia beetle microbial community structure throughout development in field and laboratory galleries__ - Analysis of Field Galleries (16S and 28S)
author: "Janina M.C. Diehl"
date: "20.10.2022"
output:
  pdf_document: default
  html_notebook: default
---

R version 4.0.5 (2021-03-31)

# __Data Preparation__

load required packages

```{r message=FALSE, warning=FALSE}
library(nlme)
library(lme4)
library(mgcv)
library(permute)
library(lattice)
library(vegan)
library(phyloseq)
library(ggplot2)
library(dplyr)
library(scales)
library(grid)
library(DHARMa)
library(ggeffects)
library(glmmTMB)
library(lmerTest)
library(emmeans)
library(sjPlot)
library(fitdistrplus)
library(GLMMadaptive)
library(microbiome)
library(microbiomeutilities)
library(knitr)
library(ggpubr)
library(doBy)
library(performance)
library(see)
library(patchwork)
library(pairwiseAdonis)
library(cowplot)
library(multcomp)
library(ggrepel)
library(car)
```

# __Load bacterial & fungal datasets and prepare phyloseq objects for analysis__ 

loading the data of bacteria

```{r}
data16S <- otu_table(read.table("16S_zotu_table_development_field.txt",sep="\t", header=T, row.names=1, check.names = F), taxa_are_rows=T)
tax16S <- tax_table(as.matrix(read.table("16S_zotus.tax_dev_field.txt", sep="\t", header=T, row.names=1, fill=T)))
datasample16S  <- sample_data(read.table("16S_map_development_field.txt", sep="\t", header=T, row.names=1))
```

loading the data of fungi

```{r}
dataLSU <- otu_table(read.table("28S_zotu_table_development_field.txt",sep="\t", header=T, row.names=1, check.names = F), taxa_are_rows=T)
taxLSU <- tax_table(as.matrix(read.table("28S_zotus.tax_development_field.txt", sep="\t", header=T, row.names=1, fill=T)))
datasampleLSU  <- sample_data(read.table("28Smap_development_field.txt", sep="\t", header=T, row.names=1))
```

## merge data into phyloseq object

```{r}
(all16S <- merge_phyloseq(data16S, tax16S, datasample16S)) 
```

```{r}
(allLSU <- merge_phyloseq(dataLSU, taxLSU, datasampleLSU)) 
```

## edit phyloseq objects

### copy taxonomic classification in tax_table collumns with gaps and add "_spc"

```{r}
all16S =  subset_taxa(all16S, (Domain == "d:Bacteria"))
tax_table(all16S)[tax_table(all16S)[,"Phylum"]=="","Phylum"]<-paste(tax_table(all16S)[tax_table(all16S)[,"Phylum"]=="","Domain"],"_spc",sep="")
tax_table(all16S)[tax_table(all16S)[,"Class"]=="","Class"]<-paste(tax_table(all16S)[tax_table(all16S)[,"Class"]=="","Phylum"],"_spc",sep="")
tax_table(all16S)[tax_table(all16S)[,"Order"]=="","Order"]<-paste(tax_table(all16S)[tax_table(all16S)[,"Order"]=="","Class"],"_spc",sep="")
tax_table(all16S)[tax_table(all16S)[,"Family"]=="","Family"]<-paste(tax_table(all16S)[tax_table(all16S)[,"Family"]=="","Order"],"_spc",sep="")
tax_table(all16S)[tax_table(all16S)[,"Genus"]=="","Genus"]<-paste(tax_table(all16S)[tax_table(all16S)[,"Genus"]=="","Family"],"_spc",sep="")
tax_table(all16S)[tax_table(all16S)[,"Species"]=="","Species"]<-paste(tax_table(all16S)[tax_table(all16S)[,"Species"]=="","Genus"],"_spc",sep="")
```

```{r}
allLSU =  subset_taxa(allLSU, (Kingdom == "k:Fungi"))
tax_table(allLSU)[tax_table(allLSU)[,"Phylum"]=="","Phylum"]<-paste(tax_table(allLSU)[tax_table(allLSU)[,"Phylum"]=="","Kingdom"],"_spc",sep="")
tax_table(allLSU)[tax_table(allLSU)[,"Class"]=="","Class"]<-paste(tax_table(allLSU)[tax_table(allLSU)[,"Class"]=="","Phylum"],"_spc",sep="")
tax_table(allLSU)[tax_table(allLSU)[,"Order"]=="","Order"]<-paste(tax_table(allLSU)[tax_table(allLSU)[,"Order"]=="","Class"],"_spc",sep="")
tax_table(allLSU)[tax_table(allLSU)[,"Family"]=="","Family"]<-paste(tax_table(allLSU)[tax_table(allLSU)[,"Family"]=="","Order"],"_spc",sep="")
tax_table(allLSU)[tax_table(allLSU)[,"Genus"]=="","Genus"]<-paste(tax_table(allLSU)[tax_table(allLSU)[,"Genus"]=="","Family"],"_spc",sep="")
tax_table(allLSU)[tax_table(allLSU)[,"Species"]=="","Species"]<-paste(tax_table(allLSU)[tax_table(allLSU)[,"Species"]=="","Genus"],"_spc",sep="")
```

### start filtering out all ZOTUs that were 

#### only assigned to Domain (Bacteria and Archaea), as well as Chloroplasts

```{r}
dataset.16S.ordi =  subset_taxa(all16S, (Domain == "d:Bacteria" | Domain =="d:Archaea"))
dataset.16S.ordi =  subset_taxa(dataset.16S.ordi, Phylum != "d:Bacteria_spc")
dataset.16S.ordi =  subset_taxa(dataset.16S.ordi, Phylum != "d:Archaea_spc")
dataset.16S.ordi =  subset_taxa(dataset.16S.ordi, Class != "c:Chloroplast")
```

#### only assigned to Kingdom (Fungi)

```{r}
dataset.LSU.ordi =  subset_taxa(allLSU, (Kingdom == "k:Fungi"))
dataset.LSU.ordi =  subset_taxa(dataset.LSU.ordi, Phylum != "k:Fungi_spc")
```

### check all the columns for patterns ranging from [a-z] joined by __ like this [a-z]__ and substitute it with "" i.e. nothing.

```{r}
tax_table(dataset.16S.ordi)[, colnames(tax_table(dataset.16S.ordi))] <- gsub(tax_table(dataset.16S.ordi)[, colnames(tax_table(dataset.16S.ordi))],     pattern = "[a-z]:", replacement = "")
tax_table(dataset.16S.ordi)[, colnames(tax_table(dataset.16S.ordi))] <- gsub(tax_table(dataset.16S.ordi)[, colnames(tax_table(dataset.16S.ordi))],     pattern = "_spc_spc_spc", replacement = "_spc")
tax_table(dataset.16S.ordi)[, colnames(tax_table(dataset.16S.ordi))] <- gsub(tax_table(dataset.16S.ordi)[, colnames(tax_table(dataset.16S.ordi))],     pattern = "_spc_spc", replacement = "_spc")
tax_table(dataset.16S.ordi)[, colnames(tax_table(dataset.16S.ordi))] <- gsub(tax_table(dataset.16S.ordi)[, colnames(tax_table(dataset.16S.ordi))],     pattern = "-", replacement = "_")
data.16S.zOTU <- dataset.16S.ordi
tax_table(dataset.16S.ordi) <- tax_table(dataset.16S.ordi)[,3:9]
```

```{r}
tax_table(dataset.LSU.ordi)[, colnames(tax_table(dataset.LSU.ordi))] <- gsub(tax_table(dataset.LSU.ordi)[, colnames(tax_table(dataset.LSU.ordi))],     pattern = "[a-z]:", replacement = "")
tax_table(dataset.LSU.ordi)[, colnames(tax_table(dataset.LSU.ordi))] <- gsub(tax_table(dataset.LSU.ordi)[, colnames(tax_table(dataset.LSU.ordi))],     pattern = "_spc_spc_spc", replacement = "_spc")
tax_table(dataset.LSU.ordi)[, colnames(tax_table(dataset.LSU.ordi))] <- gsub(tax_table(dataset.LSU.ordi)[, colnames(tax_table(dataset.LSU.ordi))],     pattern = "_spc_spc", replacement = "_spc")
tax_table(dataset.LSU.ordi)[, colnames(tax_table(dataset.LSU.ordi))] <- gsub(tax_table(dataset.LSU.ordi)[, colnames(tax_table(dataset.LSU.ordi))],     pattern = "-", replacement = "_")
data.zOTU <- dataset.LSU.ordi
tax_table(dataset.LSU.ordi) <- tax_table(dataset.LSU.ordi)[,3:9]
```

### Format the phyloseq object to add the best taxonomy in phyloseq object (tax_table and otu_table).

```{r}
dataset.16S.ordi <- format_to_besthit(dataset.16S.ordi)
taxa_names(dataset.16S.ordi)[1:5]

dataset.LSU.ordi <- format_to_besthit(dataset.LSU.ordi)
taxa_names(dataset.LSU.ordi)[1:5]
```

## check the table without Bacteria and Archaea sp. for total reads per sample

```{r}
colSums(otu_table(all16S))
colSums(otu_table(dataset.16S.ordi))

colSums(otu_table(allLSU))
colSums(otu_table(dataset.LSU.ordi))
```

### check for most abundant undefined taxa (Bacteria sp.) --> mitochondrial DNA of Fungi

```{r}
most.abundant<-subset_samples(wChl16S, Phase=="Phase1" | Phase=="Phase2" | Phase=="Phase3")
filtaxa <- names (sort(rowSums(otu_table(most.abundant)) ,decreasing=T))[1:50]
tax_table(wChl16S)[filtaxa]
round(otu_table(wChl16S)[filtaxa], digits = 4)
```

Zotu9: 	Leptographium terebrantis strain CBS 337.70 small subunit ribosomal RNA gene, partial sequence; and LAGLIDADG homing endonuclease gene, complete cds; mitochondrial 	Leptographium terebrantis 	NA 	96390 	274 	274 	100% 	3e-69 	88.60% 	3056 	GU983690.1

Zotu15:	Graphium penicillioides strain HSAUP044237 small subunit ribosomal RNA gene, partial sequence; mitochondrial 	Graphium penicillioides 	379 	379 	100% 	6e-101 	97.31% 	570 	FJ914731.1

Zotu17: 	Hyphodiscus hyaloscyphoides voucher TNS-F13588 small subunit ribosomal RNA gene, partial sequence; mitochondrial 	Hyphodiscus hyaloscyphoides 	379 	379 	100% 	6e-101 	97.31% 	925 	JN086817.1

Zotu18:   Leptographium wingfieldii strain TOM9.4 small subunit ribosomal RNA gene, partial sequence; and LAGLIDADG homing endonuclease gene, complete cds; mitochondrial 	Leptographium wingfieldii 	NA 	155675 	340 	340 	100% 	3e-89 	93.81% 	3060 	GU983696.1

Zotu32: 	Fungal sp. strain EL003500 12S ribosomal RNA gene, partial sequence; mitochondrial 	fungal sp. 	239 	239 	100% 	1e-58 	86.04% 	823 	KU354580.1

Zotu41: 	Capronia pilosella isolate AFTOL-ID 657 small subunit ribosomal RNA gene, partial sequence; mitochondrial 	Capronia pilosella 	409 	409 	100% 	7e-110 	100.00% 	716 	FJ225725.1

Zotu42: 	Togninia minima isolate AFTOL-ID 924 12S small subunit ribosomal RNA gene, partial sequence; mitochondrial 	Phaeoacremonium minimum 	NA 	223192 	388 	388 	100% 	9e-104 	98.21% 	763 	FJ190603.1

__--> Excel file with all blasted zOTUs on GitHub.__




## check controls from dataset: only ten most abundant ZOTUs are picked for visualisation

__Bacteria__
```{r}
neg.controls<-subset_samples(dataset.16S.ordi, Tree=="medium" | Tree=="negative")
sample_names(neg.controls)
```

visualisation of negative controls 

```{r}
filtaxa <- names (sort(rowSums(otu_table(neg.controls)) ,decreasing=T))[1:10] 
round(otu_table(dataset.16S.ordi)[filtaxa], digits = 4)
```
only very low read numbers of most abundanct ZOTUS in our samples
we let ZOTUS from neg. controls in the samples and choose they way of low abundance filtering!


check for ten most abundant ZOTUs in pos.controls

```{r}
pos.controls<-subset_samples(dataset.16S.ordi, Tree=="standard")
sample_names(pos.controls)
filtaxa <- names (sort(rowSums(otu_table(pos.controls)) ,decreasing=T))[1:10]
round(otu_table(dataset.16S.ordi)[filtaxa], digits = 4)
```

Zotu22: Escherichia coli O157:H7 strain 7.1_Anguil chromosome, complete genome 	Escherichia coli O157:H7 	468 	3278 	100% 	1e-127 	100.00% 	5598953 	CP076232.1

Zotu25:   Salmonella enterica subsp. enterica serovar Enteritidis strain SE211 chromosome, complete genome 	Salmonella enterica subsp. enterica serovar Enteritidis 	NA 	149539 	468 	3278 	100% 	1e-127 	100.00% 	4679414 	CP084532.1

Zotu35:   Staphylococcus aureus strain CAB4 16S ribosomal RNA gene, partial sequence 	Staphylococcus aureus 	NA 	1280 	468 	468 	100% 	1e-127 	100.00% 	1187 	OK360620.1

Zotu40:   Enterococcus faecalis strain ATCC 19433 16S ribosomal RNA gene, partial sequence 	Enterococcus faecalis 	NA 	1351 	468 	468 	100% 	1e-127 	100.00% 	1481 	OK360923.1

Zotu52:   Bacillus subtilis group sp. strain SY27.1A 16S ribosomal RNA gene and 16S-23S ribosomal RNA intergenic spacer, partial sequence 	Bacillus subtilis group sp. 	NA 	2571209 	468 	468 	100% 	1e-127 	100.00% 	1291 	OK428682.1

Zotu56:   Pseudomonas aeruginosa strain BGS1 16S ribosomal RNA gene, partial sequence 	Pseudomonas aeruginosa 	NA 	287 	468 	468 	100% 	1e-127 	100.00% 	1384 	MW857131.2

Zotu57:   Limosilactobacillus fermentum strain RAMULAB29 16S ribosomal RNA gene, partial sequence 	Limosilactobacillus fermentum 	468 	468 	100% 	1e-127 	100.00% 	1248 	OK398430.1

Zotu69:   Listeria monocytogenes strain ka89-2 16S ribosomal RNA gene, partial sequence 	Listeria monocytogenes 	NA 	1639 	468 	468 	100% 	1e-127 	100.00% 	1158 	MT633107.1

Zotu105:    Uncultured Klebsiella sp. clone p77 16S ribosomal RNA gene, partial sequence 	uncultured Klebsiella sp. 	468 	468 	100% 	1e-127 	100.00% 	766 	MF327039.1

Zotu205:    Klebsiella pneumoniae strain KPS11 16S ribosomal RNA gene, partial sequence 	Klebsiella pneumoniae 	NA 	573 	457 	457 	100% 	3e-124 	99.21% 	1432 	MN006374.1

__Fungi__ 
```{r}
neg.controls<-subset_samples(dataset.LSU.ordi, Phase=="medium" | Sample=="negative1" | Sample=="mock")
sample_names(neg.controls)
```

visualisation of negative controls 

```{r}
filtaxa <- names (sort(rowSums(otu_table(neg.controls)) ,decreasing=T))[1:10] 
round(otu_table(dataset.LSU.ordi)[filtaxa], digits = 4)
```

check for ten most abundant ZOTUs in pos.controls

```{r}
pos.controls<-subset_samples(dataset.LSU.ordi, Phase=="standard" | Sample=="negative2")
sample_names(pos.controls)
filtaxa <- names (sort(rowSums(otu_table(pos.controls)) ,decreasing=T))[1:10]
round(otu_table(dataset.LSU.ordi)[filtaxa], digits = 4)
```

## visualisation of sequencing controls

define color bar for the next plots

```{r}
Family_colors <- c("#CBD588", "burlywood1","#DA5724", "#508578", "#CD9BCD", "orange"  , "#AD6F3B",  "#673770","#0c6985","#D14285", "#652926", "#C84248", "lightgreen", "blueviolet", "darkgreen", "#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861", "#541f03",  "#b64609", "#f4671a", "#e3a380", "#d22323", "#a19b10", "#eae34e", "#6b6935", "#fff400", "#c5c398", "#5f833c" , "#9ac570", "#75d816", "#c2ff89", "#377200", "#3dc97f", "#08eccf", "#119c8b", "#77b2aa", "#3be3b4", "#237c96", "#21b2dd", "#6abed6", "#a5daea", "#0c6985", "#0c1485", "#404cf6", "#6c72c3", "#9ea2d6", "#9767c9", "#501094", "#b11dc3", "#e47ff0", "#e51bbe", "#f01c7c", "#ec84b3", "#f60838","burlywood1", "peachpuff4", "grey87", "bisque", "gold", "lightgreen", "forestgreen")
```

__Bacteria__
```{r}
negs <- subset_samples(dataset.16S.ordi, Tree=="medium" | Tree=="negative")
  
pos <- subset_samples(dataset.16S.ordi, Tree=="standard")
  
Bacteria_Genus.pos <- pos %>%               
  tax_glom(taxrank = "Genus") %>%                     
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                        
  arrange(Genus)

Bacteria_Genus.pos$Genus<-as.character(Bacteria_Genus.pos$Genus)
Bacteria_Genus.pos$Genus[Bacteria_Genus.pos$Abundance<0.01]<-"Others < 1%"
Bacteria_Genus.pos$Class<-as.character(Bacteria_Genus.pos$Class)
Bacteria_Genus.pos$Class[Bacteria_Genus.pos$Abundance<0.01]<-"Others"

Bacteria_Genus.neg <- negs %>%               
  tax_glom(taxrank = "Genus") %>%                      
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                         
  arrange(Genus)

Bacteria_Genus.neg$Genus<-as.character(Bacteria_Genus.neg$Genus)
Bacteria_Genus.neg$Genus[Bacteria_Genus.neg$Abundance<0.02]<-"Others"
Bacteria_Genus.neg$Class<-as.character(Bacteria_Genus.neg$Class)
Bacteria_Genus.neg$Class[Bacteria_Genus.neg$Abundance<0.02]<-"Others"
```

positive controls prae decontam

```{r}
Bacteria_Genus.pos_plot <-ggplot(Bacteria_Genus.pos, aes(x = Sample, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Family_colors, name = "Genus") 

g1<-Bacteria_Genus.pos_plot +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(axis.text.x=element_text(size = rel(0.5)))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))
```

negative controls prae decontam

```{r}
Bacteria_Genus.neg_plot <-ggplot(Bacteria_Genus.neg, aes(x = Sample, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Family_colors, name = "Genus") 

g2<-Bacteria_Genus.neg_plot +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(axis.text.x=element_text(size = rel(0.5)))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))

Bacteria_Genus.neg_plot2 <-ggplot(Bacteria_Genus.neg, aes(x = Phase, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Family_colors, name = "Genus")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))
```

__Fungi__
```{r}
mock <- subset_samples(dataset.LSU.ordi, Sample == "mock")
std <- subset_samples(dataset.LSU.ordi, Sample == "standard")
  
negs <- subset_samples(dataset.LSU.ordi, Tree == "medium" | Tree == "negative")
  
Fungi_Species.mock <- mock %>%               
  tax_glom(taxrank = "Species") %>%                     
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                        
  arrange(Species)

Fungi_Species.mock$Species<-as.character(Fungi_Species.mock$Species)
Fungi_Species.mock$Species[Fungi_Species.mock$Abundance<0.01]<-"Others < 1%"
Fungi_Species.mock$Class<-as.character(Fungi_Species.mock$Class)
Fungi_Species.mock$Class[Fungi_Species.mock$Abundance<0.01]<-"Others"

Fungi_Species.std <- std %>%               
  tax_glom(taxrank = "Species") %>%                     
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                        
  arrange(Species)

Fungi_Species.std$Species<-as.character(Fungi_Species.std$Species)
Fungi_Species.std$Species[Fungi_Species.std$Abundance<0.01]<-"Others < 1%"
Fungi_Species.std$Class<-as.character(Fungi_Species.std$Class)
Fungi_Species.std$Class[Fungi_Species.std$Abundance<0.01]<-"Others"

Fungi_Species.neg <- negs %>%               
  tax_glom(taxrank = "Species") %>%                      
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                         
  arrange(Species)

Fungi_Species.neg$Species<-as.character(Fungi_Species.neg$Species)
Fungi_Species.neg$Species[Fungi_Species.neg$Abundance<0.01]<-"Others"
Fungi_Species.neg$Class<-as.character(Fungi_Species.neg$Class)
Fungi_Species.neg$Class[Fungi_Species.neg$Abundance<0.01]<-"Others"
```

positive controls prae decontam

```{r}
Fungi_Species.mock_plot <-ggplot(Fungi_Species.mock, aes(x = Sample, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species") 

g1<-Fungi_Species.mock_plot +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(axis.text.x=element_text(size = rel(0.5)))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))

Fungi_Species.mock_plot2 <-ggplot(Fungi_Species.mock, aes(x = Phase, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))
```

```{r}
Fungi_Species.std_plot <-ggplot(Fungi_Species.std, aes(x = Sample, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species") 

g2<-Fungi_Species.std_plot +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(axis.text.x=element_text(size = rel(0.5)))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))

Fungi_Species.std_plot2 <-ggplot(Fungi_Species.std, aes(x = Phase, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))
```

negative controls prae decontam

```{r}
Fungi_Species.neg_plot <-ggplot(Fungi_Species.neg, aes(x = Sample, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species") 

g3<-Fungi_Species.neg_plot +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(axis.text.x=element_text(size = rel(0.5)))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))

Fungi_Species.neg_plot2 <-ggplot(Fungi_Species.neg, aes(x = Phase, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))
```

## run decontam to filter contaminating taxa

__Bacteria__
```{r}
library(decontam)
df <- as.data.frame(sample_data(dataset.16S.ordi)) # Put sample_data into a ggplot-friendly data.frame
df$LibrarySize <- sample_sums(dataset.16S.ordi)
df <- df[order(df$LibrarySize),]
df$Index <- seq(nrow(df))
ggplot(data=df, aes(x=Index, y=LibrarySize, color=Sample_or_Control)) + geom_point()
sample_data(dataset.16S.ordi)$is.neg <- sample_data(dataset.16S.ordi)$Sample_or_Control == "Control Sample"
contamdf.prev <- isContaminant(dataset.16S.ordi, method="prevalence", neg="is.neg")
table(contamdf.prev$contaminant) 
head(which(contamdf.prev$contaminant))

ps.pa <- transform_sample_counts(dataset.16S.ordi, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$Sample_or_Control == "Control Sample", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$Sample_or_Control == "True Sample", ps.pa)
```

Make data.frame of prevalence in positive and negative samples

```{r}
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),contaminant=contamdf.prev$contaminant)
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() + xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)")
ps.noncontam_dataset.16S.ordi <- prune_taxa(!contamdf.prev$contaminant, dataset.16S.ordi)
ps.noncontam_dataset.16S.ordi #277 taxa/36 samples
smin <- min(sample_sums(ps.noncontam_dataset.16S.ordi))
smean <- mean(sample_sums(ps.noncontam_dataset.16S.ordi))
smax <- max(sample_sums(ps.noncontam_dataset.16S.ordi))
cat("The minimum sample read count is:",smin) #121
cat("The average sample read count is:",smean) #20791.72
cat("The maximum sample read count is:",smax) #42604

#dat.16S.zOTU
sample_data(data.16S.zOTU)$is.neg <- sample_data(data.16S.zOTU)$Sample_or_Control == "Control Sample"
contamdf.prev2 <- isContaminant(data.16S.zOTU, method="prevalence", neg="is.neg")
table(contamdf.prev2$contaminant) #FALSE 277, TRUE 163
head(which(contamdf.prev2$contaminant))
ps.noncontam_data.16S.zOTU <- prune_taxa(!contamdf.prev2$contaminant, data.16S.zOTU)
```

__Fungi__
```{r}
df <- as.data.frame(sample_data(dataset.LSU.ordi)) # Put sample_data into a ggplot-friendly data.frame
df$LibrarySize <- sample_sums(dataset.LSU.ordi)
df <- df[order(df$LibrarySize),]
df$Index <- seq(nrow(df))
ggplot(data=df, aes(x=Index, y=LibrarySize, color=Sample_or_Control)) + geom_point()
sample_data(dataset.LSU.ordi)$is.neg <- sample_data(dataset.LSU.ordi)$Sample_or_Control == "Control Sample"
contamdf.prev <- isContaminant(dataset.LSU.ordi, method="prevalence", neg="is.neg")
table(contamdf.prev$contaminant) 
head(which(contamdf.prev$contaminant))

ps.pa <- transform_sample_counts(dataset.LSU.ordi, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$Sample_or_Control == "Control Sample", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$Sample_or_Control == "True Sample", ps.pa)
```

Make data.frame of prevalence in positive and negative samples

```{r}
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),contaminant=contamdf.prev$contaminant)
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() + xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)")
ps.noncontam_dataset.LSU.ordi <- prune_taxa(!contamdf.prev$contaminant, dataset.LSU.ordi)
ps.noncontam_dataset.LSU.ordi
smin <- min(sample_sums(ps.noncontam_dataset.LSU.ordi))
smean <- mean(sample_sums(ps.noncontam_dataset.LSU.ordi))
smax <- max(sample_sums(ps.noncontam_dataset.LSU.ordi))
cat("The minimum sample read count is:",smin) #59/6
cat("The average sample read count is:",smean) #21612.94/21059.5
cat("The maximum sample read count is:",smax) #40839/40838

#zOTU dataset
sample_data(data.zOTU)$is.neg <- sample_data(data.zOTU)$Sample_or_Control == "Control Sample"
contamdf.prev2 <- isContaminant(data.zOTU, method="prevalence", neg="is.neg")
table(contamdf.prev2$contaminant) #FALSE 279, TRUE 16 
head(which(contamdf.prev2$contaminant))
ps.noncontam_data.LSU.zOTU <- prune_taxa(!contamdf.prev2$contaminant, data.zOTU)
```

### create a list of all excluded contaminants

```{r}
contaminants <- subset(contamdf.prev, contaminant == "TRUE")
```

## check and plot controls again

__Bacteria__
```{r}
negs2 <- subset_samples(ps.noncontam_dataset.16S.ordi, Tree == "medium" | Tree == "negative")
negs2 <- prune_taxa(taxa_sums(negs2) > 0, negs2)
  
pos2 <- subset_samples(dataset.16S.ordi, Tree == "standard")
pos2 <- prune_taxa(taxa_sums(pos2) > 0, pos2)
  
Bacteria_Genus.pos2 <- pos2 %>%               
  tax_glom(taxrank = "Genus") %>%                     
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                        
  arrange(Genus)

Bacteria_Genus.pos2$Genus<-as.character(Bacteria_Genus.pos2$Genus)
Bacteria_Genus.pos2$Genus[Bacteria_Genus.pos2$Abundance<0.01]<-"Others < 1%"
Bacteria_Genus.pos2$Class<-as.character(Bacteria_Genus.pos2$Class)
Bacteria_Genus.pos2$Class[Bacteria_Genus.pos2$Abundance<0.01]<-"Others"

Bacteria_Genus.neg2 <- negs2 %>%               
  tax_glom(taxrank = "Genus") %>%                      
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                         
  arrange(Genus)

Bacteria_Genus.neg2$Genus<-as.character(Bacteria_Genus.neg2$Genus)
Bacteria_Genus.neg2$Genus[Bacteria_Genus.neg2$Abundance<0.03]<-"Others"
Bacteria_Genus.neg2$Class<-as.character(Bacteria_Genus.neg2$Class)
Bacteria_Genus.neg2$Class[Bacteria_Genus.neg2$Abundance<0.03]<-"Others"
```

positive controls post decontam

```{r}
Bacteria_Genus.pos2_plot <-ggplot(Bacteria_Genus.pos2, aes(x = Sample, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Family_colors, name = "Genus") 

g3<-Bacteria_Genus.pos2_plot +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(axis.text.x=element_text(size = rel(0.5)))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))

Bacteria_Genus.pos2_plot2 <-ggplot(Bacteria_Genus.pos2, aes(x = Phase, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Family_colors, name = "Genus")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))
```

negative controls post decontam

```{r}
Bacteria_Genus.neg2_plot <-ggplot(Bacteria_Genus.neg2, aes(x = Sample, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Family_colors, name = "Genus") 

g4<-Bacteria_Genus.neg2_plot +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(axis.text.x=element_text(size = rel(0.5)))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))

Bacteria_Genus.neg2_plot2 <-ggplot(Bacteria_Genus.neg2, aes(x = Phase, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Family_colors, name = "Genus")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))
```

__Fungi__
```{r}
mock2 <- subset_samples(ps.noncontam_dataset.LSU.ordi, Sample == "mock")
std2 <- subset_samples(ps.noncontam_dataset.LSU.ordi, Sample == "standard")
  
negs2 <- subset_samples(ps.noncontam_dataset.LSU.ordi, Tree == "negative" | Tree == "medium")
  
Fungi_Species.mock2 <- mock2 %>%               
  tax_glom(taxrank = "Species") %>%                     
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                        
  arrange(Species)

Fungi_Species.mock2$Species<-as.character(Fungi_Species.mock2$Species)
Fungi_Species.mock2$Species[Fungi_Species.mock2$Abundance<0.01]<-"Others < 1%"
Fungi_Species.mock2$Class<-as.character(Fungi_Species.mock2$Class)
Fungi_Species.mock2$Class[Fungi_Species.mock2$Abundance<0.01]<-"Others"

Fungi_Species.std2 <- std2 %>%               
  tax_glom(taxrank = "Species") %>%                     
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                        
  arrange(Species)

Fungi_Species.std2$Species<-as.character(Fungi_Species.std2$Species)
Fungi_Species.std2$Species[Fungi_Species.std2$Abundance<0.01]<-"Others < 1%"
Fungi_Species.std2$Class<-as.character(Fungi_Species.std2$Class)
Fungi_Species.std2$Class[Fungi_Species.std2$Abundance<0.01]<-"Others"

Fungi_Species.neg2 <- negs2 %>%               
  tax_glom(taxrank = "Species") %>%                      
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                         
  arrange(Species)

Fungi_Species.neg2$Species<-as.character(Fungi_Species.neg2$Species)
Fungi_Species.neg2$Species[Fungi_Species.neg2$Abundance<0.01]<-"Others"
Fungi_Species.neg2$Class<-as.character(Fungi_Species.neg2$Class)
Fungi_Species.neg2$Class[Fungi_Species.neg2$Abundance<0.01]<-"Others"
```

positive controls post decontam

```{r}
Fungi_Species.mock2_plot <-ggplot(Fungi_Species.mock2, aes(x = Sample, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species") 

g4<-Fungi_Species.mock2_plot +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(axis.text.x=element_text(size = rel(0.5)))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))

Fungi_Species.mock2_plot2 <-ggplot(Fungi_Species.mock2, aes(x = Phase, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))
```

standard controls post decontam
```{r}
Fungi_Species.std2_plot <-ggplot(Fungi_Species.std2, aes(x = Sample, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species") 

g5<-Fungi_Species.std2_plot +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(axis.text.x=element_text(size = rel(0.5)))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))

Fungi_Species.std2_plot2 <-ggplot(Fungi_Species.std2, aes(x = Phase, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))
```

negative controls post decontam

```{r}
Fungi_Species.neg2_plot <-ggplot(Fungi_Species.neg2, aes(x = Sample, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species") 

g6<-Fungi_Species.neg2_plot +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(axis.text.x=element_text(size = rel(0.5)))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))

Fungi_Species.neg2_plot2 <-ggplot(Fungi_Species.neg2, aes(x = Phase, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))
```

## exclude negative and positive samples for a look of the ten most abundant zOTUs in our samples

__Bacteria__
```{r}
bac.without.controls <- subset_samples(ps.noncontam_dataset.16S.ordi, Tree!="medium")
bac.without.controls2 <- subset_samples(bac.without.controls, Tree!="negative")
bwc <- subset_samples(bac.without.controls2, Tree!="standard")

bwc.zOTU <- subset_samples(ps.noncontam_data.16S.zOTU, Tree!="medium")
bwc.zOTU <- subset_samples(bwc.zOTU, Tree!="negative")
bwc.zOTU <- subset_samples(bwc.zOTU, Tree!="standard")
```

get rid of no read taxa in dataset
```{r}
bwc <- prune_taxa(taxa_sums(bwc) > 0, bwc)
bacF.final <- bwc

bwc.zOTU <- prune_taxa(taxa_sums(bwc.zOTU) > 0, bwc.zOTU)
```

```{r}
colSums(otu_table(bacF.final))
```
did not loose a lot of reads, still high numbers

__Fungi__
```{r}
fun.without.controls <- subset_samples(ps.noncontam_dataset.LSU.ordi, Phase!="medium")
fun.without.controls2 <- subset_samples(fun.without.controls, Phase!="negative")
fun.without.controls3 <- subset_samples(fun.without.controls2, Phase!="mock")
fwc <- subset_samples(fun.without.controls3, Phase!="standard")

fwc.zOTU <- subset_samples(ps.noncontam_data.LSU.zOTU, Phase != "medium")
fwc.zOTU <- subset_samples(fwc.zOTU, Phase != "negative")
fwc.zOTU <- subset_samples(fwc.zOTU, Phase != "mock")
fwc.zOTU <- subset_samples(fwc.zOTU, Phase != "standard")
```

get rid of no read taxa and unimportant metadata in dataset
```{r}
fwc <- prune_taxa(taxa_sums(fwc) > 0, fwc)
colSums(otu_table(fwc))
funF.final <- fwc

fwc.zOTU <- prune_taxa(taxa_sums(fwc.zOTU) > 0, fwc.zOTU)
```


## now have a look at the 30 most abundant ZOTUS in our samples

__Bacteria__
```{r}
most.abundant<-subset_samples(bacF.final, Phase=="Phase1" | Phase=="Phase2" | Phase=="Phase3")
sample_names(most.abundant)
```

```{r}
filtaxa <- names (sort(rowSums(otu_table(most.abundant)) ,decreasing=T))[1:60]
round(otu_table(bacF.final)[filtaxa], digits = 4)
```

Zotu4: Olivibacter sp. A37T4 from Portugal 16S ribosomal RNA gene, partial sequence 	Olivibacter sp. A37T4 	NA 	1713038 	451 	451 	100% 	1e-122 	98.81% 	780 	KJ654823.1

Zotu8: Sphingobacteria sp. DVSBS3 16S ribosomal RNA gene, partial sequence 	Sphingobacteriia bacterium DVSBS3 	NA 	1576838 	318 	318 	100% 	1e-82 	89.33% 	1374 	KJ004492.1

Zotu11: Ochrobactrum sp. strain CPO 4.226 16S ribosomal RNA gene, partial sequence 	Ochrobactrum sp. 	NA 	42190 	468 	468 	100% 	1e-127 	100.00% 	1368 	MN733054.1

Zotu12: Uncultured Mucilaginibacter sp. clone UPMC1 16S ribosomal RNA gene, partial sequence 	uncultured Mucilaginibacter sp. 	NA 	797541 	468 	468 	100% 	1e-127 	100.00% 	495 	KP203982.1
        Uncultured Sphingobacterium sp. clone TPMC9 16S ribosomal RNA gene, partial sequence 	uncultured Sphingobacterium sp. 	NA 	182688 	468 	468 	100% 	1e-127 	100.00% 	507 	KP203967.1
        
Zotu13: Chryseobacterium aquaticum strain C-291 16S ribosomal RNA gene, partial sequence 	Chryseobacterium aquaticum 	NA 	452084 	468 	468 	100% 	1e-127 	100.00% 	1228 	MT197478.1


__Fungi__
```{r}
most.abundant<-subset_samples(funF.final, Phase=="Phase1" | Phase=="Phase2" | Phase=="Phase3")
sample_names(most.abundant)
```

```{r}
filtaxa <- names (sort(rowSums(otu_table(most.abundant)) ,decreasing=T))[1:30]
round(otu_table(funF.final)[filtaxa], digits = 4)
```
Zotu1: Fantasmomyces hyalinus CMW 42166 28S rRNA gene, partial sequence; from TYPE material 	Fantasmomyces hyalinus 	396 	396 	96% 	7e-106 	93.04% 	812 	NG_059682.1 (also Ceratocystis in list)

Zotu2: Graphium sp. 1 RJ-2013 strain KFL15FRJTD 5.8S ribosomal RNA gene, partial sequence; internal transcribed spacer 2, complete sequence; and 28S ribosomal RNA gene, partial sequence 	Graphium sp. 1 RJ-2013 	518 	518 	100% 	1e-142 	100.00% 	908 	MH283164.1

Zotu9: Raffaelea canadensis strain CBS168.66 28S ribosomal RNA gene, partial sequence 	Raffaelea canadensis 	NA 	45347 	503 	503 	100% 	4e-138 	98.59% 	540 	EU984299.1
        Raffaelea quercina strain PC05.007 large subunit ribosomal RNA gene, partial sequence 	Raffaelea quercina 	NA 	2837398 	520 	520 	100% 	4e-143 	99.65% 	534 	JF909537.1
        
Zotu14: Raffaelea sulphurea isolate 140 28S ribosomal RNA gene, partial sequence 	Raffaelea sulphurea 	NA 	490887 	486 	486 	100% 	4e-133 	97.86% 	782 	MG673976.1

Zotu18: Graphium sp. 1 RJ-2013 strain KFL15FRJTD 5.8S ribosomal RNA gene, partial sequence; internal transcribed spacer 2, complete sequence; and 28S ribosomal RNA gene, partial sequence 	Graphium sp. 1 RJ-2013 	496 	496 	100% 	7e-136 	98.57% 	908 	MH283164.1

Zotu19: Fantasmomyces hyalinus CMW 42166 28S rRNA gene, partial sequence; from TYPE material 	Fantasmomyces hyalinus 	NA 	1925498 	379 	379 	96% 	7e-101 	91.94% 	812 	NG_059682.1 (also Ceratocystis in list)

Zotu21: Graphium sp. 2 RJ-2013 strain KFL67FRJTD 5.8S ribosomal RNA gene, partial sequence; internal transcribed spacer 2, complete sequence; and 28S ribosomal RNA gene, partial sequence 	Graphium sp. 2 RJ-2013 	NA 	2494232 	473 	473 	100% 	3e-129 	97.15% 	901 	MH283173.1

Zotu22: Raffaelea sulphurea isolate 140 28S ribosomal RNA gene, partial sequence 	Raffaelea sulphurea 	361 	361 	100% 	3e-95 	89.79% 	782 	MG673976.1

Zotu23: Raffaelea sulphurea isolate 140 28S ribosomal RNA gene, partial sequence 	Raffaelea sulphurea 	NA 	490887 	387 	387 	100% 	4e-103 	91.46% 	782 	MG673976.1

Zotu24: Phaeoacremonium austroafricanum strain CBS 112949 large subunit ribosomal RNA gene, partial sequence 	Phaeoacremonium austroafricanum 	NA 	344977 	505 	505 	100% 	1e-138 	100.00% 	895 	MH874478.1

Zotu25: Xylariales spp./ Trichoderma spp.

Zotu27: Fantasmomyces hyalinus CMW 42166 28S rRNA gene, partial sequence; from TYPE material 	Fantasmomyces hyalinus 	374 	374 	96% 	3e-99 	91.58% 	812 	NG_059682.1

Zotu29: Eutypella quaternata voucher MFLU:15-2605 large subunit ribosomal RNA gene, partial sequence 	Eutypella quaternata 	NA 	140567 	503 	503 	100% 	4e-138 	99.64% 	860 	MT183518.1

Zotu34: Raffaelea sulphurea isolate 140 28S ribosomal RNA gene, partial sequence 	Raffaelea sulphurea 	420 	420 	100% 	4e-113 	93.59% 	782 	MG673976.1

Zotu36: Raffaelea canadensis strain CBS168.66 28S ribosomal RNA gene, partial sequence 	Raffaelea canadensis 	NA 	45347 	497 	497 	100% 	2e-136 	98.24% 	540 	EU984299.1

Zotu38: Raffaelea fusca isolate MAR23 large subunit ribosomal RNA gene, partial sequence 	Raffaelea fusca 	NA 	490883 	353 	353 	100% 	4e-93 	89.44% 	814 	MN012888.1

Zotu40: Xylariales spp. or 	Raffaelea sulphurea isolate 140 28S ribosomal RNA gene, partial sequence 	Raffaelea sulphurea 	339 	339 	100% 	1e-88 	88.38% 	782 	MG673976.1

Zotu57: Raffaelea aff. canadensis C2711 26S ribosomal RNA gene, partial sequence 	Raffaelea aff. canadensis C2711 	448 	448 	100% 	2e-121 	95.39% 	549 	HQ688665.1

Zotu90: Fantasmomyces hyalinus CMW 42166 28S rRNA gene, partial sequence; from TYPE material 	Fantasmomyces hyalinus 	NA 	1925498 	340 	340 	96% 	3e-89 	89.38% 	812 	NG_059682.1

### have a look at the whole datasets

```{r}
Bacteria_Genus.all. <- ps.noncontam_dataset.16S.ordi %>%               
  tax_glom(taxrank = "Genus") %>%                      
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                         
  filter(Abundance > 0.05) %>%                        
  arrange(Genus)

Bacteria_Genus.all.$Phase <- factor(Bacteria_Genus.all.$Phase, levels = c("Phase1", "Phase2", "Phase3", "medium", "negative", "standard"))

Bacteria_Genus_plot <-ggplot(Bacteria_Genus.all., aes(x = Sample, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Family_colors, name = "Genus") 

Bacteria_Genus_plot +
  facet_wrap("Phase", scales = "free_x",  drop = TRUE)+
  theme(plot.title = element_text(size = 16, face = "bold")) + 
  theme(text = element_text(size=16, face = "bold"))+                         
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+      theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x="Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(axis.text.x=element_text(angle = 90, hjust = 1, size = rel(1)))+
  theme(legend.text = element_text(face = "italic")) 
```
Most abundant taxa are _Sphingobacterium, Pseudoxanthomonas, Pedobacter & Erwinia_. 

```{r}
Fungi_Species.all. <- ps.noncontam_dataset.LSU.ordi %>%               
  tax_glom(taxrank = "Species") %>%                      
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                         
  filter(Abundance > 0.05) %>%                        
  arrange(Species)

Fungi_Species.all.$Phase <- factor(Fungi_Species.all.$Phase, levels = c("Phase1", "Phase2", "Phase3", "medium", "negative", "standard", "mock"))

Fungi_Species_plot <-ggplot(Fungi_Species.all., aes(x = Sample, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Family_colors, name = "Species") 

p <- Fungi_Species_plot +
  facet_wrap("Phase", scales = "free_x",  drop = TRUE)+
  theme(plot.title = element_text(size = 16, face = "bold")) + 
  theme(text = element_text(size=16, face = "bold"))+                         
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+      theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x="Sample", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(axis.text.x=element_text(angle = 90, hjust = 1, size = rel(1)))+
  theme(legend.text = element_text(face = "italic"))
```

# __Analysis of bacterial community__

## Rarefaction curves 

1. all samples with controls
2. all samples with controls decontaminated
3. all samples without controls


Rarefaction is used to simulate even number of reads per sample.

```{r}
rarecurve(t(otu_table(dataset.16S.ordi)), cex=0.6, step = 20, main = "Rarefaction curve of all samples (incl. controls)")
rarecurve(t(otu_table(ps.noncontam_dataset.16S.ordi)), cex=0.6, step = 20, main = "Rarefaction curve of all samples (incl. controls) after decontamination")
rarecurve(t(otu_table(bacF.final)), cex=0.6, step = 20, main = "Rarefaction curve of samples (excl. controls)")
```


rarefy to 10000 reads
```{r}
ps2 <- bacF.final
set.seed(1)
ps2 <- rarefy_even_depth(ps2,sample.size=10000, replace=FALSE, rngseed = 1)
rarecurve(t(otu_table(ps2)), cex=0.6, step = 20)
bacR <- ps2
```
--> 2OTUs were removed because they are no longer present in any sample after random subsampling


## __Alpha diversity__

### Diversity plots

This returns a table with selected diversity indicators. Check a separate page on Alpha for other functions.

```{r}
tab <- microbiome::alpha(bacR, index = c("diversity_shannon","observed"))
```


#### Prepare data for vizualisation

Now, get the metadata (sample_data) from the phyloseq object

```{r}
ps1.meta <- meta(bacR)
```

Add the diversity table to metadata

```{r}
ps1.meta$Shannon <- tab$diversity_shannon 
ps1.meta$Observed <- tab$observed
```

create a list of pairwise comparisons
```{r}
phase <- unique(ps1.meta$Phase) # get the variables
```

make a pairwise list that we want to compare.

```{r}
phase.pairs <- combn(seq_along(phase), 2, simplify = FALSE, FUN = function(i)phase[i])

print(phase.pairs)
```

### plot

ps2 (rarefied, 10000 reads)
```{r}
ps1.meta$Phase <- as.factor(ps1.meta$Phase)
ps1.meta$Phase <- factor(ps1.meta$Phase, levels = c("Phase1", "Phase2", "Phase3"), labels = c("eggs & larvae", "larvae, pupae & adults", "adults"))
```

__Shannon diversity index__

```{r}
##Phase
p1 <- ggplot(ps1.meta, aes(x = Phase, y = Shannon))

p1 <- p1+ geom_boxplot(colour = "brown4", size = 0.9, alpha = 0.2, width = 0.5)+
    geom_jitter(position = position_jitter(w=0.1))+
    theme_bw()+
    labs(x = "Developmental Stage", y = "Shannon's Diversity Index")+
  theme(axis.title = element_text(size = 18, face = "bold"), axis.text = element_text(size = 16))
```

add statistical output to plot
```{r}
shan_bacFP <- p1 + 
        annotate("text", label = c("a", "a", "a"), y = 3.1, x = c("eggs & larvae", "larvae, pupae & adults", "adults"), size = 7)+
  ylim(NA, 3.3)
```

__Observed richness__

```{r}
p2 <- ggplot(ps1.meta, aes(x = Phase, y = Observed))

p2 <- p2+ geom_boxplot(colour = "brown4", size = 0.9, alpha = 0.2, width = 0.5)+
    geom_jitter(position = position_jitter(w=0.1))+
    theme_bw()+
    labs(x = "Developmental Phase", y = "Observed richness")+
  theme(axis.title = element_text(size = 18, face = "bold"), axis.text = element_text(size = 16))
```

add statistical output to plot
```{r}
ob_bacFP <- p4 + 
        annotate("text", label = c("a", "a", "a"), y = 112, x = c("eggs & larvae", "larvae, pupae & adults", "adults"), size = 7)+
  ylim(NA, 115)
```

### Statistics

Testing differences in alpha diversity

create dataframe for analysis
```{r}
d <- meta(bacR)
shannon <- diversity(bacR, "shannon")
d$shannon <- shannon$shannon
observed <- alpha(bacR, index = "observed", zeroes = TRUE)
d$observed <- observed$observed
```

__Shannon diversity__

test distribution
```{r}
hist(d$shannon)
descdist(d$shannon, boot = 1000)
fit.lnorm <- fitdist(d$shannon, distr = "lnorm", method = "mme") # looks good
plot(fit.lnorm)
fit.norm <- fitdist(d$shannon, distr = "norm", method = "mme")
plot(fit.norm)
```
--> normal distribution

run model
```{r} 
d$Phase <- as.factor(d$Phase)
d <- within(d, Treatment <- relevel(Phase, ref = "Phase1"))

shan <- glmmTMB(shannon ~ Phase + (1|Tree), data = d) 
summary(shan) 
Anova(shan, type = "II")
```

test model (Performance & Dharma)
```{r}
check_model(shan)

res_shan <- simulateResiduals(shan, n = 1000)
testResiduals(res_shan)
plotResiduals(res_shan)
```

post-hoc test displaying the result table with summary()
```{r}
post.hoc <- emmeans(shan, pairwise ~ Phase, adjust = "tukey")
summary(post.hoc)
```

__Observed richness__ 

```{r}
hist(d$observed)
descdist(d$observed, boot = 1000)
fit.norm <- fitdist(d$observed, distr = "norm", method = "mme")
plot(fit.norm)
```

run model
```{r}
ob <- glmmTMB(observed ~ Phase + (1|Tree), data = d) 
summary(ob) 
Anova(ob, type = "II")
```

test model
```{r}
check_model(ob)

res_ob <- simulateResiduals(ob, n = 1000)
testResiduals(res_ob)
plotResiduals(res_ob)
```

post-hoc test displaying the result table with summary()
```{r}
post.hoc <- emmeans(ob, pairwise ~ Phase, adjust = "tukey")
summary(post.hoc)
```

## __Beta Diversity__

transform total read counts into proportional/compositional data
```{r}
rel.bacF <- bacF.final %>%               
  transform_sample_counts(function(x) {x/sum(x)} )
```


### relative Abundance plots

agglomerate data to Genus level, transform to rel. abundance, melt long format and sort dataframe alph. by Genus

```{r, warning=F}
Bacteria_Genus <- bacF.final %>%               
  tax_glom(taxrank = "Genus") %>%                     
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                        
  arrange(Genus)
```

find the mean and standard deviation by Genus

```{r}
mean.bwc <- tapply(Bacteria_Genus$Abundance, Bacteria_Genus$Genus, mean)
mean.bwc

phase1 <- subset(Bacteria_Genus, Phase == "Phase1")
mean.ph1 <- tapply(phase1$Abundance, phase1$Genus, mean)
mean.ph1
SD.ph1 <- tapply(phase1$Abundance, phase1$Genus, sd)
SD.ph1

phase2 <- subset(Bacteria_Genus, Phase == "Phase2")
mean.ph2 <- tapply(phase2$Abundance, phase2$Genus, mean)
mean.ph2
SD.ph2 <- tapply(phase2$Abundance, phase2$Genus, sd)
SD.ph2

phase3 <- subset(Bacteria_Genus, Phase == "Phase3")
mean.ph3 <- tapply(phase3$Abundance, phase3$Genus, mean)
mean.ph3
SD.ph3 <- tapply(phase3$Abundance, phase3$Genus, sd)
SD.ph3

SD.bwc <- tapply(Bacteria_Genus$Abundance, Bacteria_Genus$Genus, sd)
SD.bwc
```

edit object for plotting

```{r}
Bacteria_Genus$Genus<-as.character(Bacteria_Genus$Genus)
Bacteria_Genus$Genus[Bacteria_Genus$Abundance<0.05]<-"Others"
Bacteria_Genus$Class<-as.character(Bacteria_Genus$Class)
Bacteria_Genus$Class[Bacteria_Genus$Abundance<0.05]<-"Others"
Bacteria_Genus$Genus<-factor(Bacteria_Genus$Genus)
Bacteria_Genus<-droplevels(Bacteria_Genus)         
Bacteria_Genus$Phase<-factor(Bacteria_Genus$Phase,levels=c("Phase1", "Phase2", "Phase3")) 
Bacteria_Genus$Genus<-factor(Bacteria_Genus$Genus,levels=levels(Bacteria_Genus$Genus))
Bacteria_Genus$Genus <- factor(Bacteria_Genus$Genus, levels = c("Others", "Pseudochrobactrum", "Ochrobactrum", "Phyllobacterium", "Burkholderia", "Xanthomonas", "Pseudoxanthomonas", "Erwinia", "Flavobacterium", "Chryseobacterium", "Brachybacterium", "Demetria", "Mucilaginibacter", "Taibaiella", "Olivibacter", "Pedobacter", "Bacteroidetes_spc", "Sphingobacterium", "Sphingobacteriia_spc"), labels = c("Others (< 0.5%)", "Pseudochrobactrum", "Ochrobactrum", "Phyllobacterium", "Burkholderia", "Xanthomonas", "Pseudoxanthomonas", "Erwinia", "Flavobacterium", "Chryseobacterium", "Brachybacterium", "Demetria", "Mucilaginibacter", "Taibaiella", "Olivibacter", "Pedobacter", "Bacteroidetes spp.", "Sphingobacterium", "Sphingobacteriia spp."))
```

define colour bar
```{r}
Plot_colors_g <- c("grey87", "firebrick", "indianred1", "red", "lavenderblush", "green2", "seagreen", "palegreen", "khaki1", "yellow1", "skyblue", "slategray1", "navajowhite3", "burlywood4", "saddlebrown", "sandybrown", "bisque3", "bisque4", "bisque2")
```


__Phase__

```{r}
Bacteria_Genus_plot <-ggplot(Bacteria_Genus, aes(x = Phase, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Plot_colors_g, name = "Genus") 

bacF1<-Bacteria_Genus_plot +
  theme(text = element_text(size=25, face = "bold"))+                          
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 13))+
  theme(legend.text = element_text(face = "italic"))+
  theme(axis.text.y=element_text(size = 15))+
  theme(axis.title.y = element_text(size=20, "bold"))+
  theme(axis.title.x = element_blank())+ 
  theme(axis.text.x = element_blank())
```

__Sample/Phase__

```{r}
Bacteria_Genus_plot2 <-ggplot(Bacteria_Genus, aes(x = Sample, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position="fill") +
  scale_fill_manual(values = Plot_colors_g, name = "Genus")

g2<-Bacteria_Genus_plot2 +
  facet_wrap("Phase", scales = "free_x",  drop = TRUE)+
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Samples", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))+
  theme(axis.text.x=element_text(size = rel(0.9), angle = 90))
```

### Microbiome composition

#### Composition heatmap

```{r}
bac.heat <- ggplot(Bacteria_Genus, aes(Sample, Genus)) + geom_tile(aes(fill = Abundance)) +
                  facet_grid(~Phase, scales = "free_x", space = "free_x", drop = TRUE)

bac.heat <- bac.heat + scale_fill_distiller("Abundance", palette = "RdYlBu", labels=scales::percent) + theme_bw() 

bac.heat <- bac.heat + theme(axis.text.y = element_text(colour = 'black', 
                                                    size = 13, 
                                                    face = 'italic')) +
                        theme(axis.title.y = element_text(size=17, "bold")) +
                        theme(legend.title = element_text(size = 15), legend.text = element_text(size = 13))+
  theme(axis.title.x = element_blank())+
  theme(axis.ticks.x = element_blank())+
  theme(axis.text.x = element_blank())

bac.heat <- bac.heat + ylab("Genus")

bac.heat <- bac.heat + theme(legend.key = element_blank(), 
                     strip.background = element_rect(colour="black", fill="white"))

bac.heat <- bac.heat + scale_y_discrete(position = "right") 

bac.heat <- bac.heat + theme(legend.position = "left")

bac.heat2 <- bac.heat + theme(strip.text.x = element_blank())
```

#### Core taxa abundance plot

relative abundance 

```{r}
colpal_bac2 <- c("seagreen", "palegreen", "red", "yellow1", "sandybrown", "saddlebrown", "bisque4")
```

```{r relative abundance}
ps4 <- prune_taxa(taxa_sums(rel.bacF) > 0, rel.bacF)
ps4 <- tax_glom(ps4, taxrank = 'Genus')
psOrd4 = subset_taxa(ps4, Genus=="Pseudoxanthomonas" | Genus=="Erwinia" | Genus=="Phyllobacterium" | Genus=="Chryseobacterium" | Genus=="Pedobacter" | Genus=="Olivibacter" | Genus=="Sphingobacterium")
psPh1 = subset_samples(psOrd4, Phase=="Phase1")
#Melt and plot
melt<-psmelt(psOrd4)
melt2 <- psmelt(psPh1)
levels2=c("Phase1","Phase2","Phase3")
change <- c("Phase1","Phase2","Phase3")
melt$Genus <- factor(melt$Genus, levels = c("Pseudoxanthomonas", "Erwinia", "Phyllobacterium", "Chryseobacterium", "Pedobacter", "Olivibacter", "Sphingobacterium"))

a_mean <- melt2 %>%
  group_by(Genus) %>%
  summarize(mean_val = mean(Abundance))
print(a_mean)

p<-ggplot(data = melt, aes(x = Phase, y = Abundance)) +
  geom_boxplot(aes(fill=Genus),alpha=0.5,lwd=0.7, position = position_dodge(width = 0.3), width=0.45,outlier.shape = NA) + geom_jitter(aes(color = Genus), height = 0, width = .1)+
  geom_hline(data= a_mean, aes(yintercept=mean_val), width=5,size=1.2,linetype="dashed", col = "darkred")+
  scale_fill_manual(values = colpal_bac2, name = "Genus")+
  scale_color_manual(values = colpal_bac2, name = "Genus")+
  labs(x = "", y = "Abundance\n")+
  facet_grid(factor(Genus, levels = c("Pseudoxanthomonas", "Erwinia", "Phyllobacterium", "Chryseobacterium", "Pedobacter", "Olivibacter", "Sphingobacterium"))~Phase, scales = "free")+theme_bw()
 
p<-p+ theme(legend.position="right")+ylab("Relative abundance")
p<-p+ theme(legend.text=element_text(size=15, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),legend.key.size = unit(0.9, "cm"))+theme(legend.title = element_text(size = 20, face = "bold"))
abu<-p +theme(strip.text.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))+
  scale_y_continuous(labels=percent_format(), limits = c(NA, 1))
Phase<-abu+theme(axis.title.y = element_text(size=20))+theme(axis.text.y = element_text(size=15, face="bold")) +
  #theme(axis.text.x = element_text(size=20, angle=90,face="bold"))+
  theme(axis.title.x = element_blank())+
  theme(axis.title.y = element_blank())+
  theme(axis.text.x = element_blank())+
  theme(axis.ticks.x = element_blank())+ 
  theme(strip.text.y = element_blank())

relAb_bacF <- Phase + theme(panel.spacing = unit(1.5, "lines"))
```

### Ordination analysis

__NMDS__

```{r}
set.seed(1)
ordi = ordinate(rel.bac, "NMDS", "bray", k=3, trymax=100)
ordi$stress #0.140176 --> fair
```
As a rule of thumb, an NMDS ordination with a stress value around or above 0.2 is deemed suspect and a stress value approaching 0.3 indicates that the ordination is arbitrary. Stress values equal to or below 0.1 are considered fair, while values equal to or below 0.05 indicate good fit. Allowing the algorithm to ordinate in more dimensions can reduce the stress; however, allowing more than 3 dimensions quickly makes interpretation more challenging.

```{r}
NMDS_bacF <- plot_ordination(rel.bac, ordi, color = "Phase", shape = "Tree", axes=c(1,2))+
  stat_ellipse(level = 0.95, geom = "polygon", aes(fill = Phase), alpha = 0.25)+
  scale_shape_manual(values=c(16, 17, 8, 15))+
  geom_point(size=3, inherit.aes=T)+
  annotate(geom="text", x=-1.2, y= -1.5, label="stress = 0.140",
              color="black", size = 6)+
  theme_bw()+
  theme(legend.justification=c(1,0), legend.position=c(1,0))+
  theme(axis.text = element_text(size = 14, face = "bold"))+
  theme(axis.title = element_text(size = 18, face = "bold"))+
  theme(legend.text = element_text(size = 12))+
  theme(legend.title = element_text(size = 16))+
  theme(title = element_text(size = 18))

NMDS_bacF
```

### Permanova for community-level multivariate comparisons

PERMANOVA quantifies multivariate community-level differences between groups.
Pick relative abundances (compositional) and sample metadata
```{r}
pseq.rel <- microbiome::transform(bacF.final, "compositional")
otu <- abundances(pseq.rel)
meta <- meta(pseq.rel)
```

#### PERMANOVA significance test for group-level differences

Now let us evaluate whether the 'Developmental Phase' has a significant effect on overall garden microbiota composition. Perform PERMANOVA:

```{r}
set.seed(1)
adonis(distance(pseq.rel, method = "bray") ~ Phase * Tree, data = meta)

set.seed(1)
adonis2(distance(pseq.rel, method = "bray") ~ Phase + Tree, data = meta)

set.seed(1)
pairwise.adonis2(distance(pseq.rel, method = "bray") ~ Phase + Tree, data = meta)
```

### PERMDISP: test of homogeneity of dispersion
This test can be done to see if one group has more compositional variance than another. Moreover, homogeneity of dispersion among groups is very advisable to have if you want to test if two or more groups have different compositions, which is tested by adonis.

```{r}
dist <- vegdist(t(otu))
dist2 <- betadisper(dist, meta$Phase)
anova(dist2)
permutest(dist2, pairwise = TRUE, permutations = 99)
(disp.HSD <- TukeyHSD(dist2))
plot(dist2, hull = FALSE, ellipse = TRUE)

dist3 <- betadisper(dist, meta$Tree)
anova(dist3)
permutest(dist3, pairwise = TRUE, permutations = 99)
(disp.HSD <- TukeyHSD(dist3))
plot(dist3, hull = FALSE, ellipse = TRUE)
```

--> result is heavily influenced not by the difference in composition between groups but by differences in composition within groups (heterogeneous dispersion, and thus a measure of betadiversity).


# __Analysis fungal community__

## Rarefaction curves 

1. all samples with controls
2. all samples with controls after decontam
3. all samples without controls

Rarefaction is used to simulate even number of reads per sample.

```{r}
rarecurve(t(otu_table(dataset.LSU.ordi)), cex=0.6, step = 20, main = "Rarefaction curve of all samples (incl. controls)")
rarecurve(t(otu_table(ps.noncontam_dataset.LSU.ordi)), cex=0.6, step = 20, main = "Rarefaction curve of all samples (incl. controls) after decontamination")
rarecurve(t(otu_table(funF.final)), cex=0.6, step = 20, main = "Rarefaction curve of samples (excl. controls)")
```

## __Alpha diversity__

rarefy to 4000 reads
```{r}
ps2 <-funF.final
set.seed(1)
ps2 <- rarefy_even_depth(ps2,sample.size=4000, replace=FALSE, rngseed = 1)
rarecurve(t(otu_table(ps2)), cex=0.6, step = 20)
funR <- ps2
```
--> 32OTUs were removed because they are no longer present in any sample after random sub-sampling

### Diversity plots

#### Prepare data for vizualisation

```{r}
tab <- microbiome::alpha(funR, index = c("diversity_shannon","observed"))
```

Now, get the metadata (sample_data) from the phyloseq object

```{r}
ps1.meta <- meta(funR)
```

Add the diversity table to metadata

```{r}
ps1.meta$Shannon <- tab$diversity_shannon 
ps1.meta$Observed <- tab$observed
```

create a list of pairwise comparisons
```{r}
phase <- unique(ps1.meta$Phase) # get the variables
```

make a pairwise list that we want to compare.

```{r}
phase.pairs <- combn(seq_along(phase), 2, simplify = FALSE, FUN = function(i)phase[i])
print(phase.pairs)
```

#### plot

ps2 (rarefied, 2000 reads)
```{r}
ps1.meta$Phase <- as.factor(ps1.meta$Phase)
ps1.meta$Phase <- factor(ps1.meta$Phase, levels = c("Phase1", "Phase2", "Phase3"), labels = c("eggs & larvae", "larvae, pupae & adults", "adults"))
```

__Shannon diversity index__
```{r}
p1 <- ggplot(ps1.meta, aes(x = Phase, y = Shannon))
p1 <- p1+ geom_boxplot(colour = "brown4", size = 0.9, alpha = 0.2, width = 0.5)+
    geom_jitter(position = position_jitter(w=0.1))+
    theme_bw()+
    labs(x = "Developmental Stage", y = "Shannon's Diversity Index")+
  theme(axis.title = element_text(size = 18, face = "bold"), axis.text = element_text(size = 16))
```

add statistical data output to plot
```{r}
shan_funFP <- p1 + 
        annotate("text", label = c("a", "ab", "b"), y = 3.1, x = c("eggs & larvae", "larvae, pupae & adults", "adults"), size = 7)+
  ylim(NA, 3.2)
```

__Observed richness__
```{r}
p4b <- ggplot(ps1.meta, aes(x = Phase, y = Observed))
p4b <- p4b+ geom_boxplot(colour = "brown4", size = 0.9, alpha = 0.2, width = 0.5)+
    geom_jitter(position = position_jitter(w=0.1))+
    theme_bw()+
    labs(x = "Developmental Phase", y = "Observed richness")+
  theme(axis.title = element_text(size = 18, face = "bold"), axis.text = element_text(size = 16))
```

add statistical data output to plot
```{r}
ob_funFP <- p4b + 
        annotate("text", label = c("a", "a", "a"), y = 114, x = c("eggs & larvae", "larvae, pupae & adults", "adults"), size = 7)+
  ylim(NA, 115)
```

### __Statistics__

Testing differences in alpha diversity

Create the dataframe for analysis
```{r}
d <- meta(funR)
shannon <- diversity(funR, "shannon")
d$shannon <- shannon$shannon
observed <- alpha(funR, index = "observed", zeroes = TRUE)
d$observed <- observed$observed
```

__Shannon diversity__
```{r}
hist(d$shannon)
descdist(d$shannon, boot = 1000)
fit.gamma <- fitdist(d$shannon, distr = "gamma", method = "mme") # looks good
plot(fit.gamma)
fit.norm <- fitdist(d$shannon, distr = "norm", method = "mme")
plot(fit.norm)
```
--> normal distribution

```{r}
d$Phase <- as.factor(d$Phase)
d$Tree <- as.factor(d$Tree)
d <- within(d, Phase <- relevel(Phase, ref = "Phase1"))

shan <- glmmTMB(shannon ~ Phase + (1|Tree), data = d) 
summary(shan) #AIC = 46.5
Anova(shan, type = "II")
```

test model
```{r}
check_model(shan)

res_shan <- simulateResiduals(shan, n = 1000)
testResiduals(res_shan)
plotResiduals(res_shan)
```

```{r}
post.hoc <- emmeans(shan1, pairwise ~ Phase, adjust = "tukey")
summary(post.hoc)
```

__Observed richness__
```{r}
hist(d$observed)
descdist(d$observed, boot = 1000)
fit.gamma <- fitdist(d$observed, distr = "gamma", method = "mme") # looks good
plot(fit.gamma)
fit.norm <- fitdist(d$observed, distr = "norm", method = "mme")
plot(fit.norm)
```
--> normal distribution

```{r}
ob <- glmmTMB(observed ~ Phase + (1|Tree), data = d) 
summary(ob)
Anova(ob, type = "II")
```

test model
```{r}
check_model(ob)

res_ob <- simulateResiduals(ob, n = 1000)
testResiduals(res_ob)
plotResiduals(res_ob)
```


## Beta Diversity

transform total read counts into proportional/compositional data
```{r}
rel.funF <- funF.final %>%               
  transform_sample_counts(function(x) {x/sum(x)} )
```

### relative Abundance plots

agglomerate data to Species level, transform to rel. abundance, melt long format and sort dataframe alph. by Species

```{r}
Fungi_Species <- funF.final %>%               
  tax_glom(taxrank = "Species") %>%                     
  transform_sample_counts(function(x) {x/sum(x)} ) %>% 
  psmelt() %>%                                        
  arrange(Species) 
```

find the mean and standard deviation by Species

```{r}
mean.fwc <- tapply(Fungi_Species$Abundance, Fungi_Species$Species, mean)
mean.fwc

phase1 <- subset(Fungi_Species, Phase == "Phase1")
mean.phase1 <- tapply(phase1$Abundance, phase1$Species, mean)
mean.phase1
SD.phase1 <- tapply(phase1$Abundance, phase1$Species, sd)
SD.phase1

phase2 <- subset(Fungi_Species, Phase == "Phase2")
mean.phase2 <- tapply(phase2$Abundance, phase2$Species, mean)
mean.phase2
SD.phase2 <- tapply(phase2$Abundance, phase2$Species, sd)
SD.phase2

phase3 <- subset(Fungi_Species, Phase == "Phase3")
mean.phase3 <- tapply(phase3$Abundance, phase3$Species, mean)
mean.phase3
SD.phase3 <- tapply(phase3$Abundance, phase3$Species, sd)
SD.phase3

SD.fwc <- tapply(Fungi_Species$Abundance, Fungi_Species$Species, sd)
SD.fwc
```

#### edit object for plotting

```{r}
Fungi_Species$Species<-as.character(Fungi_Species$Species)
Fungi_Species$Species[Fungi_Species$Abundance<0.05]<-"Others"
Fungi_Species$Class<-as.character(Fungi_Species$Class)
Fungi_Species$Class[Fungi_Species$Abundance<0.05]<-"Others"
Fungi_Species$Species<-factor(Fungi_Species$Species)
Fungi_Species<-droplevels(Fungi_Species)         
Fungi_Species$Phase<-factor(Fungi_Species$Phase,levels=c("Phase1", "Phase2", "Phase3")) 
Fungi_Species$Species<-factor(Fungi_Species$Species,levels=levels(Fungi_Species$Species))
Fungi_Species$Species <- factor(Fungi_Species$Species, levels = c("Others", "Pyxidiophora_arvernensis", "Orbiliaceae_spc", "Phaeoacremonium_spc", "Phaeoacremonium_austroafricanum", "Eutypella_quaternata", "Eutypella_or_Selenosporella_spc", "Helotiaceae_spc", "Acremonium_Monocillium_spc", "Herpotrichiellaceae_spc", "Capronia_pilosella",  "Nectria_balansae", "Clonostachys_spc", "Clonostachys_rosea", "Trichoderma_deliquescens", "Neonectria_spc", "Graphium_spc", "Raffaelea_aff._canadensis", "Raffaelea_canadensis", "Raffaelea_sulphurea", "Sordariomycetes_spc"), labels = c("Others (< 0.5%)", "Pyxidiophora arvernensis", "Orbiliaceae spp.", "Phaeoacremonium sp.", "Phaeoacremonium austroafricanum", "Eutypella quaternata", "Eutypella or Selenosporella sp.", "Helotiaceae spp.", "Acremonium/Monocillium sp.", "Herpotrichiellaceae spp.", "Capronia pilosella",  "Nectria balansae", "Clonostachys sp.", "Clonostachys rosea", "Trichoderma deliquescens", "Neonectria sp.", "Graphium sp.", "Raffaelea aff. canadensis", "Raffaelea canadensis", "Raffaelea sulphurea", "Sordariomycetes spp.")) 
```

#### define plot colours 

```{r}
Plot_colors_s <- c("grey87", "deeppink3", "powderblue", "darkseagreen2", "darkseagreen", "cyan3", "cyan", "lavenderblush3", "peachpuff3", "lightpink", "lightcoral", "tan1", "darkorange4", "darkorange3", "orange","sienna1", "lightcyan", "goldenrod3", "goldenrod3", "gold", "ivory2")
```

__Phase__
```{r}
Fungi_Species_plot <-ggplot(Fungi_Species, aes(x = Phase, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Plot_colors_s, name = "Species") 

funF1<-Fungi_Species_plot +
  theme(text = element_text(size=25, face = "bold"))+                          
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 13))+
  theme(legend.text = element_text(face = "italic"))+
  theme(axis.text.y=element_text(size = 15))+
  theme(axis.title.y = element_text(size=20, "bold"))+
  theme(axis.title.x = element_blank())+ 
  theme(axis.text.x = element_blank())
```


__Sample/Phase__
```{r}
Fungi_Species_plot2 <-ggplot(Fungi_Species, aes(x = Sample, y = Abundance, fill = Species)) + 
  geom_bar(stat = "identity", color = NA, position="fill") +
  scale_fill_manual(values = Plot_colors_s, name = "Species")

g2<-Fungi_Species_plot2 +
  facet_wrap("Phase", scales = "free_x",  drop = TRUE)+
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(text = element_text(size=20, face = "bold"))+                          
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 14))+   
  theme_classic()+           #gets rid of background
  theme(panel.grid.major = element_line(colour = "grey50"))+
  labs(x=" Samples", y="Relative Abundance")+
  scale_y_continuous(labels=percent_format())+
  theme(legend.text = element_text(face = "italic"))+
  theme(axis.text.x=element_text(size = rel(0.9), angle = 90))
```


### Microbiome composition

#### Composition heatmap

```{r}
fun.heat <- ggplot(Fungi_Species, aes(Sample, Species)) + geom_tile(aes(fill = Abundance)) +
                  facet_grid(~Phase, scales = "free_x", space = "free_x", drop = TRUE)

fun.heat <- fun.heat + scale_fill_distiller("Abundance", palette = "RdYlBu", labels=scales::percent) + theme_bw() 

fun.heat <- fun.heat + theme(axis.text.y = element_text(colour = 'black', 
                                                    size = 13, 
                                                    face = 'italic')) +
                        theme(axis.text.x = element_text(colour = 'black', size = 13, angle = 90))+
                        theme(axis.title = element_text(size=17, "bold")) +
                        theme(legend.title = element_text(size = 15), legend.text = element_text(size = 13))

fun.heat <- fun.heat + ylab("Species")

fun.heat <- fun.heat + theme(legend.key = element_blank(), 
                     strip.background = element_rect(colour="black", fill="white"))

fun.heat <- fun.heat + scale_y_discrete(position = "right") 

fun.heat <- fun.heat + theme(legend.position = "left")

fun.heat2 <- fun.heat + theme(strip.text.x = element_blank())
```

combined bac + fun
```{r}
pseqB <- aggregate_rare(rel.bac, level = "Species", detection = 5/100, prevalence = 10/100)
pseqF <- aggregate_rare(rel.fun, level = "Species", detection = 5/100, prevalence = 10/100)

pseqBh <- pseqB %>% 
  psmelt()
pseqBh$Group <- "Bacteria"
pseqFh <- pseqF %>% 
  psmelt()
pseqFh$Group <- "Fungi"

pseqh <- bind_rows(pseqBh, pseqFh)

pseqh$Phase<-factor(pseqh$Phase,levels=c("Phase1", "Phase2", "Phase3"))
pseqh$Group<-factor(pseqh$Group,levels=c("Bacteria", "Fungi"))
pseqh$Species<-factor(pseqh$Species,levels=c("Pseudoxanthomonas_spadix", "Erwinia_spc", "Phyllobacterium_catacumbae","Ochrobactrum_spc", "Olivibacter_spc", "Sphingobacterium_detergens", "Sphingobacterium_spc", "Sphingobacteriia_spc", "Pedobacter_spc", "Chryseobacterium_xinjiangense", "Other", "Sordariomycetes_spc", "Raffaelea_sulphurea", "Raffaelea_canadensis", "Graphium_spc", "Clonostachys_rosea", "Other"), labels = c("Pseudoxanthomonas spadix", "Erwinia sp.", "Phyllobacterium catacumbae","Ochrobactrum sp.", "Olivibacter sp.", "Sphingobacterium detergens", "Sphingobacterium sp.", "Sphingobacteriia spp.", "Pedobacter sp.", "Chryseobacterium xinjiangense", "Other (<5%)", "Sordariomycetes spp", "Raffaelea sulphurea", "Raffaelea canadensis", "Graphium sp.", "Clonostachys rosea", "Other (<5%)"))

p.heat <- ggplot(pseqh, aes(x = Sample, y = Species)) + geom_tile(aes(fill = Abundance)) 

# Change color
p.heat <- p.heat + scale_fill_distiller("Abundance", palette = "RdYlBu", labels = scales::percent) + theme_bw() 

# Make bacterial names italics
p.heat <- p.heat + theme(axis.text.y = element_text(colour = 'black', 
                                                    size = 10, 
                                                    face = 'italic')) 
# Make separate samples based on main variable
p.heat <- p.heat + facet_grid(Group ~ Phase, scales = "free", space = "free") + rremove("x.text") 

#Clean the x-axis
p.heat <- p.heat + theme(axis.title.x=element_blank(),
                     axis.text.x=element_blank(),
                     axis.ticks.x=element_blank())

# Clean the facet label box
p.heat <- p.heat + theme(legend.key = element_blank(), 
                     strip.background = element_rect(colour="black", fill="white"))+
                    theme(strip.text = element_text(size = 15))

p.heat <- p.heat + theme(axis.text.x=element_text(angle = 90, size = 9))+
  theme(axis.title.y = element_text(size = 18, face = "bold"), axis.title.x = element_blank())+
  theme(axis.text.y = element_text(size = 13, face = "italic"))+
  theme(legend.text = element_text(size = 13))+
  theme(legend.title = element_text(size = 18, face = "bold"))
```

### Core visualization

#### Core taxa abundance

rel. abundance

```{r}
colpal_fun3 <- c("gold", "goldenrod3", "lightcyan3", "ivory2")
```

```{r relative abundance}
ps4 <- prune_taxa(taxa_sums(rel.funF) > 0, rel.funF)
ps4 <- tax_glom(ps4, taxrank = 'Species')
psOrd4 = subset_taxa(ps4, Species=="Raffaelea_sulphurea" | Species=="Raffaelea_canadensis" | Species=="Graphium_spc" | Species=="Sordariomycetes_spc")
psPh1 = subset_samples(psOrd4, Phase=="Phase1")
#Melt and plot
melt<-psmelt(psOrd4)
melt2 <- psmelt(psPh1)
levels2=c("Phase1","Phase2","Phase3")
change <- c("Phase1","Phase2","Phase3")
melt$Species <- factor(melt$Species, levels = c("Raffaelea_sulphurea", "Raffaelea_canadensis", "Graphium_spc", "Sordariomycetes_spc"), labels = c("Raffaelea sulphurea", "Raffaelea canadensis", "Graphium sp.", "Sordariomycetes sp."))
melt2$Species <- factor(melt2$Species, levels = c("Raffaelea_sulphurea", "Raffaelea_canadensis", "Graphium_spc", "Sordariomycetes_spc"), labels = c("Raffaelea sulphurea", "Raffaelea canadensis", "Graphium sp.", "Sordariomycetes sp."))

a_mean <- melt2 %>%
  group_by(Species) %>%
  summarize(mean_val = mean(Abundance))
print(a_mean)

p<-ggplot(data = melt, aes(x = Phase, y = Abundance)) +
  geom_boxplot(aes(fill=Species),alpha=0.5,lwd=0.7, position = position_dodge(width = 0.3), width=0.45,outlier.shape = NA) + geom_jitter(aes(color = Species), height = 0, width = .1)+geom_hline(data= a_mean, aes(yintercept=mean_val), width=5,size=1.2,linetype="dashed", col = "darkred")+
  scale_fill_manual(values = colpal_fun3, name = "Species")+
  scale_color_manual(values = colpal_fun3, name = "Species")+
  labs(x = "", y = "Abundance\n")+
  facet_grid(Species~Phase, scales = "free")+theme_bw()#factor(Species, levels = c("Raffaelea sulphurea", "Raffaelea canadensis", "Graphium sp", "Sordariomycetes sp."))
  
p<-p+ theme(legend.position="right")+ylab("Relative abundance")
p<-p+ theme(legend.text=element_text(size=15, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),legend.key.size = unit(0.9, "cm"))+theme(legend.title = element_text(size = 20, face = "bold"))
abu<-p +theme(strip.text.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))+
  scale_y_continuous(labels=percent_format(), limits = c(NA, 1))
Phase<-abu+theme(axis.title.y = element_text(size=20))+theme(axis.text.y = element_text(size=15, face="bold")) +
  #theme(axis.text.x = element_text(size=20, angle=90,face="bold"))+
  theme(axis.title.x = element_blank())+
  theme(axis.title.y = element_blank())+
  theme(axis.text.x = element_blank())+
  theme(axis.ticks.x = element_blank())+ 
  theme(strip.text.y = element_blank())
relAb_funF <- Phase + theme(panel.spacing = unit(1.5, "lines"))
```

### Ordination analysis

__NMDS__

```{r}
set.seed(1)
ordi1 = ordinate(rel.fun, "NMDS", "bray", k=3, trymax=100)
ordi1$stress #0.09086746 --> ok
```

Phase
```{r}
NMDS_funF <- plot_ordination(rel.fun, ordi1, color = "Phase", shape = "Tree", axes=c(1,2))+
  stat_ellipse(level = 0.95, geom = "polygon", aes(fill = Phase), alpha = 0.25)+
  scale_shape_manual(values=c(16, 17, 8, 15))+
  geom_point(size=3, inherit.aes=T)+
  annotate(geom="text", x=-1.4, y= -1.5, label="stress = 0.091",
              color="black", size = 6)+
  theme_bw()+
  xlim(NA, 1.7)+
  theme(legend.justification=c(1,0), legend.position=c(1,0))+
  theme(axis.text = element_text(size = 14, face = "bold"))+
  theme(axis.title = element_text(size = 18, face = "bold"))+
  theme(legend.text = element_text(size = 12))+
  theme(legend.title = element_text(size = 16))+
  theme(title = element_text(size = 18))

NMDS_funF
```

### Permanova for community-level multivariate comparisons

PERMANOVA quantifies multivariate community-level differences between groups.
Pick relative abundances (compositional) and sample metadata 
```{r}
pseq.rel <- microbiome::transform(funF.final, "compositional")
otu <- abundances(pseq.rel)
meta <- meta(pseq.rel)
```


#### PERMANOVA significance test for group-level differences

Now let us evaluate whether the Treatment has a significant effect on overall garden microbiota composition. Perform PERMANOVA:

```{r}
set.seed(1)
adonis(distance(pseq.rel, method = "bray") ~ Phase * Tree, data = meta)

set.seed(1)
adonis2(distance(pseq.rel, method = "bray") ~ Phase + Tree, data = meta)

set.seed(1)
pairwise.adonis2(distance(pseq.rel, method = "bray") ~ Phase + Tree, data = meta)
```

#### Checking the homogeneity condition

Check that variance homogeneity assumptions hold (to ensure the reliability of the results):

```{r}
disp <- vegdist(t(otu))
disp2 <- betadisper(disp, meta$Phase)
anova(disp2)

permutest(disp2, pairwise = TRUE, permutations = 99)

plot(disp2, hull = FALSE, ellipse = TRUE)

disp3 <- betadisper(disp, meta$Tree)
anova(disp2)

permutest(disp3, pairwise = TRUE, permutations = 99)

plot(disp3, hull = FALSE, ellipse = TRUE)
```

# testing the fungal core taxa abundance

subset dataset Fungi_Species of both combinations to most abundant Species (each Sordariomycetes (unknown), *Graphium*, *Raffaelea canadensis* & *Raffaelea sulphurea*) 
```{r}
coretax <- subset(Fungi_Species, (Species == "Sordariomycetes_spc" | Species == "Graphium_spc" | Species == "Raffaelea_canadensis" | Species == "Raffaelea_sulphurea"))
```

```{r}
rel.coretax <- coretax

S <- subset(rel.coretax, (Species == "Sordariomycetes_spc"))
G <- subset(rel.coretax, (Species == "Graphium_spc"))
RS <- subset(rel.coretax, (Species == "Raffaelea_sulphurea"))
RC <- subset(rel.coretax, (Species == "Raffaelea_canadensis"))
```

__Sordariomycetes (relative)__
```{r}
S$Phase <- as.factor(S$Phase)
S <- within(S, Phase <- relevel(Phase, ref = "Phase1"))

#plot

S$Phase <- factor(S$Phase, levels = c("Phase1", "Phase2", "Phase3"))

SP <- ggboxplot(S, y = "rel.Abundance_Sample", x = "Phase", fill = "Species") +
        scale_fill_manual(values=c("ivory2"))+
        geom_jitter()+
        theme_bw()+
        theme(strip.text.x=element_text(size=14, face="bold"))

SP <- ggpar(SP, ylab = "Relative Abundance (%)", xlab = "Developmental Phase")+
            rremove("legend")+
            #font("title", size = 15)+
            font("xy.text", size = 14)+
            font("xy.title", size = 16, face = "bold")
SP
```

```{r}
hist(S$rel.Abundance_Sample)

descdist(S$rel.Abundance_Sample, boot = 1000)
fit.beta <- fitdist(CG$Abundance_Sample, distr = "beta", method = "mme")
plot(fit.beta)
fit.norm <- fitdist(S$rel.Abundance_Sample, distr = "norm", method = "mme")
plot(fit.norm)
fit.gamma <- fitdist(CG$rel.Abundance_Sample, distr = "gamma", method = "mme")
plot(fit.gamma)


S.mod <- glmmTMB(rel.Abundance_Sample ~ Phase + (1|Tree), data = S) 
Anova(S.mod, type = "II")

check_model(S.mod)

S.mod2 <- lme(rel.Abundance_Sample ~ Phase, random=~1|Tree, data = S) #outliers!
Anova(S.mod2, type = "II")
compare_performance(S.mod2, S.mod, rank = TRUE) #S.mod2 better

emmeans(S.mod2, pairwise ~ Phase, adjust = "tukey")
```

check for variance structure
```{r}
S$rel.Abundance_SampleTuk <- rcompanion::transformTukey(S$rel.Abundance_Sample)

m <- lme(rel.Abundance_Sample ~ Phase,random=~1|Tree,control = lmeControl(opt="optim"),data=S)
Anova(m2)
m1 <- lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree,control = lmeControl(opt="optim"),data=S)
m2 <- gls(rel.Abundance_SampleTuk ~ Phase,control = lmeControl(opt="optim"),data=S)
m3 <- lm(rel.Abundance_Sample ~ Phase,data=S)
AIC(m, m1, m2, m3) #m1 has lower AIC --> include rf

MODEL<-m
par(mfrow=c(2,2))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="response"),lpars=list(col="red",lty=1))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
qqnorm(residuals(MODEL,type="response"));qqline(residuals(MODEL,type="response"))
qqnorm(unlist(ranef(MODEL))); qqline(unlist(ranef(MODEL)))
par(mfrow=c(1,1))
scatter.smooth(S$rel.Abundance_Sample,residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
boxplot(residuals(MODEL,type="normalized")~S$Phase,drop=TRUE)
```

Variance differs between Phase and possibly Development
```{r}
vf1<-varIdent(form=~1|Phase)
vf2<-varIdent(form=~1|Tree)
vf3<-varIdent(form=~1|Phase * Tree)

m.1<-lme(rel.Abundance_Sample ~ Phase,random=~1|Tree, weights=vf1, control = lmeControl(opt="optim"), data=S)
m.2<-lme(rel.Abundance_Sample ~ Phase,random=~1|Tree, weights=vf2, control = lmeControl(opt="optim"), data=S)
m.3<-lme(rel.Abundance_Sample ~ Phase,random=~1|Tree, weights=vf3, control = lmeControl(opt="optim"), data=S)
 
AIC(m,m.1,m.2,m.3) # m lowest AIC
anova(m,m.1,m.2,m.3) 
compare_performance(m,m.1,m.2,m.3, rank = TRUE) #best m.1

Anova(m.1)

par(mfrow=c(2,2))
plot(m, which = c(1), add.smooth = FALSE)
plot(m.1, which = c(1), add.smooth = FALSE)
```


STEP 3: Choose transformation if necessary
```{r}
par(mfrow=c(2,2));hist(S$rel.Abundance_Sample);hist(sqrt(S$rel.Abundance_Sample));hist(S$rel.Abundance_SampleTuk);par(mfrow=c(1,1))
m1<-lme(rel.Abundance_Sample~ Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=S)
m2<-lme(sqrt(rel.Abundance_Sample)~ Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=S)
m3<-lme(rel.Abundance_SampleTuk~ Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=S)
AIC(m1, m2, m3) #m2 lowest AIC
Anova(m2)

MODEL<-m2
par(mfrow=c(2,2))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="response"),lpars=list(col="red",lty=1))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
qqnorm(residuals(MODEL,type="response"));qqline(residuals(MODEL,type="response"))
qqnorm(unlist(ranef(MODEL))); qqline(unlist(ranef(MODEL)))
par(mfrow=c(1,1))
```

post hoc
```{r}
HSD_Ph <- emmeans(m2, pairwise ~ Phase, adjust = "tukey")
```

create compact letters
```{r}
marginal_Ph = emmeans(m2, ~ Phase)
CLD.S <- cld(marginal_Ph, alpha = 0.05, Letters = letters, adjust = "tukey") ### Use lower-case letters for .group, Tukey-adjusted comparisons
```

extracting the compact letter display and editing groups
```{r}
cldPh <- as.data.frame.list(CLD.S)
cldPh$Phase <- factor(cldPh$Phase, levels = c("Phase1", "Phase2", "Phase3"))

S_funF <- SP + geom_text(data = cldPh, aes(y = 105, label = trimws(.group)), size = 6, show.legend = FALSE, colour = "black")
```

__*R. sulphurea* (relative)__
```{r}
RS$Phase <- as.factor(RS$Phase)
RS <- within(RS, Phase <- relevel(Phase, ref = "Phase1"))

#plot
RS$Phase <- factor(RS$Phase, levels = c("Phase1", "Phase2", "Phase3"))
RSP <- ggboxplot(RS, y = "rel.Abundance_Sample", x = "Phase", fill = "Species") +
        #ggtitle("rel. abundance of Raffaelea sulphurea in all samples for Development Phase") +
        scale_fill_manual(values=c("gold"))+
        geom_jitter()+
        theme_bw()+
        theme(strip.text.x=element_text(size=14, face="bold"))

RSP <- ggpar(RSP, ylab = "Relative Abundance (%)", xlab = "Development Phase")+
            rremove("legend")+
            #font("title", size = 15)+
            font("xy.text", size = 14)+
            font("xy.title", size = 16, face = "bold")
RSP
```

```{r}
hist(RS$rel.Abundance_Sample)

descdist(RS$rel.Abundance_Sample, boot = 1000)
fit.beta <- fitdist(RS$Abundance_Sample, distr = "beta", method = "mme")
plot(fit.beta)
fit.norm <- fitdist(RS$Abundance_Sample, distr = "norm", method = "mme")
plot(fit.norm)
fit.gamma <- fitdist(RS$rel.Abundance_Sample, distr = "gamma", method = "mme")
plot(fit.gamma)

RS.mod <- lme(rel.Abundance_Sample ~ Phase, random = ~1|Tree, data = RS) #has outliers? and significant deviation
Anova(RS.mod, type = "II")

RS.mod2 <- glmmTMB(rel.Abundance_Sample ~ Phase + (1|Tree), data = RS)
Anova(RS.mod2)

compare_performance(RS.mod, RS.mod2, rank = TRUE)
AIC(RS.mod, RS.mod2)

res_RS <- simulateResiduals(fittedModel=RS.mod2, n = 1000)
testResiduals(res_RS)
plotResiduals(res_RS)

emmeans(RS.mod2, pairwise ~ Phase, adjust = "tukey")
```

check for variance structure
```{r}
RS$rel.Abundance_SampleTuk <- rcompanion::transformTukey(RS$rel.Abundance_Sample) # lambda = 0.15

m <- lme(rel.Abundance_Sample ~ Phase,random=~1|Tree,control = lmeControl(opt="optim"),data=RS)
Anova(m3)
m1 <- lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree,control = lmeControl(opt="optim"),data=RS)
m2 <- gls(rel.Abundance_SampleTuk ~ Phase,control = lmeControl(opt="optim"),data=RS)
m3 <- lm(rel.Abundance_SampleTuk ~ Phase, control = lmeControl(opt="optim"),data=RS)
AIC(m, m1, m2, m3) #m1 has lower AIC --> include rf

MODEL<-m1
par(mfrow=c(2,2))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="response"),lpars=list(col="red",lty=1))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
qqnorm(residuals(MODEL,type="response"));qqline(residuals(MODEL,type="response"))
qqnorm(unlist(ranef(MODEL))); qqline(unlist(ranef(MODEL)))
par(mfrow=c(1,1))
scatter.smooth(RS$rel.Abundance_Sample,residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
boxplot(residuals(MODEL,type="normalized")~RS$Phase,drop=TRUE)
```

Variance differs between Phase
```{r}
vf1<-varIdent(form=~1|Phase)
vf2<-varIdent(form=~1|Tree)
vf3<-varIdent(form=~1|Phase * Tree)


m1.1<-lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree, weights=vf1, control = lmeControl(opt="optim"), data=RS)
m1.2<-lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree, weights=vf2, control = lmeControl(opt="optim"), data=RS)
m1.3<-lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree, weights=vf3, control = lmeControl(opt="optim"), data=RS)


AIC(m1,m1.1,m1.2,m1.3) #m1 lowest AIC
anova(m1,m1.1,m1.2,m1.3) #lowest m1.6
compare_performance(m1,m1.1,m1.2,m1.3, rank = TRUE) #best m1

Anova(m1)

par(mfrow=c(2,2))
plot(m1, which = c(1), add.smooth = FALSE)
plot(m1.2, which = c(1), add.smooth = FALSE)
plot(m1., which = c(1), add.smooth = FALSE)
```

STEP 3: Choose transformation if necessary
```{r}
par(mfrow=c(2,2));hist(RS$rel.Abundance_Sample);hist(sqrt(RS$rel.Abundance_Sample));hist(RS$rel.Abundance_SampleTuk);par(mfrow=c(1,1))
m1<-lme(rel.Abundance_Sample~Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=RS)
m2<-lme(sqrt(rel.Abundance_Sample)~Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=RS)
m3<-lme(rel.Abundance_SampleTuk~Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=RS)
AIC(m1, m2, m3) #m3 lowest AIC
Anova(m3)

MODEL<-m3
par(mfrow=c(2,2))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="response"),lpars=list(col="red",lty=1))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
qqnorm(residuals(MODEL,type="response"));qqline(residuals(MODEL,type="response"))
qqnorm(unlist(ranef(MODEL))); qqline(unlist(ranef(MODEL)))
par(mfrow=c(1,1))
```

post hoc
```{r}
HSD_Phase <- emmeans(m3, pairwise ~ Phase, adjust = "tukey")
```

create compact letters
```{r}
marginal_Ph = emmeans(m3, ~ Phase)
CLD.RS <- cld(marginal_Ph, alpha = 0.05, Letters = letters, adjust = "tukey")
```

extracting the compact letter display and editing groups
```{r}
cldPh <- as.data.frame.list(CLD.RS)
cldPh$Phase <- factor(cldPh$Phase, levels = c("Phase1", "Phase2", "Phase3"))

RS_fun_Ph <- RSP + geom_text(data = cldPh, aes(y = 30, label = trimws(.group)), size = 6, show.legend = FALSE, colour = "black")
```


__*R. canadensis* (relative)__
```{r}
RC$Phase <- as.factor(RC$Phase)
RC <- within(RC, Phase <- relevel(Phase, ref = "Phase1"))

# plot
RC$Phase <- factor(RC$Phase, levels = c("Phase1", "Phase2", "Phase3"))
RCP <- ggboxplot(RC, y = "rel.Abundance_Sample", x = "Phase", fill = "Species") +
        scale_fill_manual(values=c("goldenrod"))+
        geom_jitter()+
        theme_bw()+
        theme(strip.text.x=element_text(size=14, face="bold"))

RCP <- ggpar(RCP, ylab = "Relative Abundance (%)", xlab = "Development Speed")+
            rremove("legend")+
            #font("title", size = 15)+
            font("xy.text", size = 14)+
            font("xy.title", size = 16, face = "bold")
RCP
```

```{r}
hist(RC$rel.Abundance_Sample)

descdist(RC$rel.Abundance_Sample, boot = 1000)
fit.beta <- fitdist(RC$Abundance_Sample, distr = "beta", method = "mme")
plot(fit.beta)
fit.gamma <- fitdist(RC$rel.Abundance_Sample, distr = "gamma", method = "mme")
plot(fit.gamma)

RC.mod <- glmmTMB(rel.Abundance_Sample ~ Phase + (1|Tree), data = RC, family = tweedie) #better
summary(RC.mod)
Anova(RC.mod, type = "II")

check_model(RC.mod)

res_RC <- simulateResiduals(fittedModel=RC.mod, n = 1000)
testResiduals(res_RC)
plotResiduals(res_RC)

emmeans(RC.mod, pairwise ~ Phase, adjust = "tukey")
```

check for variance structure
```{r}
RC$rel.Abundance_SampleTuk <- rcompanion::transformTukey(RC$rel.Abundance_Sample) # lambda = 0.15

m <- lme(rel.Abundance_Sample ~ Phase,random=~1|Tree,control = lmeControl(opt="optim"),data=RC)
Anova(m2)
m1 <- lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree,control = lmeControl(opt="optim"),data=RC)
m2 <- gls(rel.Abundance_SampleTuk ~ Phase,control = lmeControl(opt="optim"),data=RC)
m3 <- lm(rel.Abundance_SampleTuk ~ Phase, control = lmeControl(opt="optim"),data=RC)
AIC(m, m1, m2, m3) #m3 has lower AIC --> include rf to go same  way like other models

MODEL<-m1
par(mfrow=c(2,2))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="response"),lpars=list(col="red",lty=1))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
qqnorm(residuals(MODEL,type="response"));qqline(residuals(MODEL,type="response"))
qqnorm(unlist(ranef(MODEL))); qqline(unlist(ranef(MODEL)))
par(mfrow=c(1,1))
scatter.smooth(RC$rel.Abundance_Sample,residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
boxplot(residuals(MODEL,type="normalized")~RC$Phase,drop=TRUE)
```

Variance differs between Phase
```{r}
vf1<-varIdent(form=~1|Phase)
vf2<-varIdent(form=~1|Tree)
vf3<-varIdent(form=~1|Phase * Tree)


m1.1<-lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree, weights=vf1, control = lmeControl(opt="optim"), data=RC)
m1.2<-lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree, weights=vf2, control = lmeControl(opt="optim"), data=RC)
m1.3<-lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree, weights=vf3, control = lmeControl(opt="optim"), data=RC)


AIC(m1,m1.1,m1.2,m1.3) #m1 lowest AIC
anova(m1,m1.1,m1.2,m1.3) #lowest m1
compare_performance(m1,m1.1,m1.2,m1.3, rank = TRUE) #best m1

Anova(m1)

par(mfrow=c(2,2))
plot(m1, which = c(1), add.smooth = FALSE)
plot(m2, which = c(1), add.smooth = FALSE)
```

STEP 3: Choose transformation if necessary
```{r}
par(mfrow=c(2,2));hist(RC$rel.Abundance_Sample);hist(sqrt(RC$rel.Abundance_Sample));hist(RC$rel.Abundance_SampleTuk);par(mfrow=c(1,1))
m1<-lme(rel.Abundance_Sample~Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=RC)
m2<-lme(sqrt(rel.Abundance_Sample)~Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=RC)
m3<-lme(rel.Abundance_SampleTuk~Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=RC)
AIC(m1, m2, m3) #m3 lowest AIC
Anova(m3)

MODEL<-m3
par(mfrow=c(2,2))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="response"),lpars=list(col="red",lty=1))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
qqnorm(residuals(MODEL,type="response"));qqline(residuals(MODEL,type="response"))
qqnorm(unlist(ranef(MODEL))); qqline(unlist(ranef(MODEL)))
par(mfrow=c(1,1))
```

post hoc
```{r}
HSD_Phase <- emmeans(m3, pairwise ~ Phase, adjust = "tukey")

#create compact letters
marginal_Ph = emmeans(m3, ~ Phase)
CLD.RS <- cld(marginal_Ph, alpha = 0.05, Letters = letters, adjust = "tukey")
```

```{r}
# extracting the compact letter display and editing groups
cldPh <- as.data.frame.list(CLD.RS)
cldPh$Phase <- factor(cldPh$Phase, levels = c("Phase1", "Phase2", "Phase3"))

RC_fun_Ph <- RCP + geom_text(data = cldPh, aes(y = 50, label = trimws(.group)), size = 6, show.legend = FALSE, colour = "black")
```

__*Graphium* (relative)__
```{r}
G$Phase <- as.factor(G$Phase)
G <- within(G, Phase <- relevel(Phase, ref = "Phase1"))

#plot

G$Phase <- factor(G$Phase, levels = c("Phase1", "Phase2", "Phase3"))

GP <- ggboxplot(G, y = "rel.Abundance_Sample", x = "Phase", fill = "Species") +
        scale_fill_manual(values=c("lightcyan"))+
        geom_jitter()+
        theme_bw()+
        theme(strip.text.x=element_text(size=14, face="bold"))

GP <- ggpar(GP, ylab = "Relative Abundance (%)", xlab = "Developmental Phase")+
            rremove("legend")+
            #font("title", size = 15)+
            font("xy.text", size = 14)+
            font("xy.title", size = 16, face = "bold")
GP
```

```{r}
hist(G$rel.Abundance_Sample)

descdist(G$rel.Abundance_Sample, boot = 1000)
fit.gamma <- fitdist(G$rel.Abundance_Sample, distr = "gamma", method = "mme")
plot(fit.gamma)

G.mod <- glmmTMB(rel.Abundance_Sample ~ Phase + (1|Tree), data = G, family = tweedie) 
Anova(G.mod, type = "II")

check_model(G.mod)

res_G <- simulateResiduals(fittedModel=G.mod, n = 1000)
testResiduals(res_G)
plotResiduals(res_G)

emmeans(G.mod, pairwise ~ Phase, adjust = "tukey")
```

check for variance structure
```{r}
G$rel.Abundance_SampleTuk <- rcompanion::transformTukey(G$rel.Abundance_Sample) # lambda = 0.15

m <- lme(rel.Abundance_Sample ~ Phase,random=~1|Tree,control = lmeControl(opt="optim"),data=G)
Anova(m3)
m1 <- lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree,control = lmeControl(opt="optim"),data=G)
m2 <- gls(rel.Abundance_SampleTuk ~ Phase,control = lmeControl(opt="optim"),data=G)
m3 <- lm(rel.Abundance_SampleTuk ~ Phase, control = lmeControl(opt="optim"),data=G)
AIC(m, m1, m2, m3) #m3 has lower AIC --> include rf to go same  way like other models
compare_performance(m1, m2, m3, rank = TRUE)

MODEL<-m1
par(mfrow=c(2,2))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="response"),lpars=list(col="red",lty=1))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
qqnorm(residuals(MODEL,type="response"));qqline(residuals(MODEL,type="response"))
qqnorm(unlist(ranef(MODEL))); qqline(unlist(ranef(MODEL)))
par(mfrow=c(1,1))
scatter.smooth(G$rel.Abundance_Sample,residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
boxplot(residuals(MODEL,type="normalized")~G$Phase,drop=TRUE)
```

Variance differs between Phase
```{r}
vf1<-varIdent(form=~1|Phase)
vf2<-varIdent(form=~1|Tree)
vf3<-varIdent(form=~1|Phase * Tree)


m1.1<-lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree, weights=vf1, control = lmeControl(opt="optim"), data=G)
m1.2<-lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree, weights=vf2, control = lmeControl(opt="optim"), data=G)
m1.3<-lme(rel.Abundance_SampleTuk ~ Phase,random=~1|Tree, weights=vf3, control = lmeControl(opt="optim"), data=G)


AIC(m1,m1.1,m1.2,m1.3) #m1 lowest AIC
anova(m1,m1.1,m1.2,m1.3) #lowest m1
compare_performance(m1,m1.1,m1.2,m1.3, rank = TRUE) #best m1.1

Anova(m1.1)

par(mfrow=c(2,2))
plot(m1, which = c(1), add.smooth = FALSE)
plot(m1.1, which = c(1), add.smooth = FALSE)
```

STEP 3: Choose transformation if necessary
```{r}
par(mfrow=c(2,2));hist(G$rel.Abundance_Sample);hist(sqrt(G$rel.Abundance_Sample));hist(G$rel.Abundance_SampleTuk);par(mfrow=c(1,1))
m1<-lme(rel.Abundance_Sample~Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=G)
m2<-lme(sqrt(rel.Abundance_Sample)~Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=G)
m3<-lme(rel.Abundance_SampleTuk~Phase, random = ~1|Tree, control = lmeControl(opt="optim"), data=G)
AIC(m1, m2, m3) #m3 lowest AIC
Anova(m3)

MODEL<-m3
par(mfrow=c(2,2))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="response"),lpars=list(col="red",lty=1))
scatter.smooth(fitted(MODEL),residuals(MODEL,type="normalized"),lpars=list(col="red",lty=1))
qqnorm(residuals(MODEL,type="response"));qqline(residuals(MODEL,type="response"))
qqnorm(unlist(ranef(MODEL))); qqline(unlist(ranef(MODEL)))
par(mfrow=c(1,1))
```

post hoc
```{r}
HSD_Phase <- emmeans(m3, pairwise ~ Phase, adjust = "tukey")

#create compact letters
marginal_Ph = emmeans(m3, ~ Phase)
CLD.G <- cld(marginal_Ph, alpha = 0.05, Letters = letters, adjust = "tukey")
```

extracting the compact letter display and editing groups
```{r}
cldPh <- as.data.frame.list(CLD.G)
cldPh$Phase <- factor(cldPh$Phase, levels = c("Phase1", "Phase2", "Phase3"))

G_fun_Ph <- GP + geom_text(data = cldPh, aes(y = 45, label = trimws(.group)), size = 6, show.legend = FALSE, colour = "black")
```
